// secretjs/dist/protobuf/cosmos/app/v1alpha1/config.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.178.0
//   protoc               v3.21.3
// source: cosmos/app/v1alpha1/config.proto
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GolangBinding = exports.ModuleConfig = exports.Config = exports.protobufPackage = void 0;
/* eslint-disable */
const _m0 = __importStar(require("protobufjs/minimal"));
const any_1 = require("../../../google/protobuf/any");
exports.protobufPackage = "cosmos.app.v1alpha1";
function createBaseConfig() {
    return { modules: [], golang_bindings: [] };
}
exports.Config = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.modules) {
            exports.ModuleConfig.encode(v, writer.uint32(10).fork()).ldelim();
        }
        for (const v of message.golang_bindings) {
            exports.GolangBinding.encode(v, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConfig();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.modules.push(exports.ModuleConfig.decode(reader, reader.uint32()));
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.golang_bindings.push(exports.GolangBinding.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            modules: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.modules)
                ? object.modules.map((e) => exports.ModuleConfig.fromJSON(e))
                : [],
            golang_bindings: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.golang_bindings)
                ? object.golang_bindings.map((e) => exports.GolangBinding.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        var _a, _b;
        const obj = {};
        if ((_a = message.modules) === null || _a === void 0 ? void 0 : _a.length) {
            obj.modules = message.modules.map((e) => exports.ModuleConfig.toJSON(e));
        }
        if ((_b = message.golang_bindings) === null || _b === void 0 ? void 0 : _b.length) {
            obj.golang_bindings = message.golang_bindings.map((e) => exports.GolangBinding.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.Config.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseConfig();
        message.modules = ((_a = object.modules) === null || _a === void 0 ? void 0 : _a.map((e) => exports.ModuleConfig.fromPartial(e))) || [];
        message.golang_bindings = ((_b = object.golang_bindings) === null || _b === void 0 ? void 0 : _b.map((e) => exports.GolangBinding.fromPartial(e))) || [];
        return message;
    },
};
function createBaseModuleConfig() {
    return { name: "", config: undefined, golang_bindings: [] };
}
exports.ModuleConfig = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.name !== "") {
            writer.uint32(10).string(message.name);
        }
        if (message.config !== undefined) {
            any_1.Any.encode(message.config, writer.uint32(18).fork()).ldelim();
        }
        for (const v of message.golang_bindings) {
            exports.GolangBinding.encode(v, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseModuleConfig();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.name = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.config = any_1.Any.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.golang_bindings.push(exports.GolangBinding.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            name: isSet(object.name) ? globalThis.String(object.name) : "",
            config: isSet(object.config) ? any_1.Any.fromJSON(object.config) : undefined,
            golang_bindings: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.golang_bindings)
                ? object.golang_bindings.map((e) => exports.GolangBinding.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        var _a;
        const obj = {};
        if (message.name !== "") {
            obj.name = message.name;
        }
        if (message.config !== undefined) {
            obj.config = any_1.Any.toJSON(message.config);
        }
        if ((_a = message.golang_bindings) === null || _a === void 0 ? void 0 : _a.length) {
            obj.golang_bindings = message.golang_bindings.map((e) => exports.GolangBinding.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.ModuleConfig.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseModuleConfig();
        message.name = (_a = object.name) !== null && _a !== void 0 ? _a : "";
        message.config = (object.config !== undefined && object.config !== null)
            ? any_1.Any.fromPartial(object.config)
            : undefined;
        message.golang_bindings = ((_b = object.golang_bindings) === null || _b === void 0 ? void 0 : _b.map((e) => exports.GolangBinding.fromPartial(e))) || [];
        return message;
    },
};
function createBaseGolangBinding() {
    return { interface_type: "", implementation: "" };
}
exports.GolangBinding = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.interface_type !== "") {
            writer.uint32(10).string(message.interface_type);
        }
        if (message.implementation !== "") {
            writer.uint32(18).string(message.implementation);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGolangBinding();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.interface_type = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.implementation = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            interface_type: isSet(object.interface_type) ? globalThis.String(object.interface_type) : "",
            implementation: isSet(object.implementation) ? globalThis.String(object.implementation) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.interface_type !== "") {
            obj.interface_type = message.interface_type;
        }
        if (message.implementation !== "") {
            obj.implementation = message.implementation;
        }
        return obj;
    },
    create(base) {
        return exports.GolangBinding.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseGolangBinding();
        message.interface_type = (_a = object.interface_type) !== null && _a !== void 0 ? _a : "";
        message.implementation = (_b = object.implementation) !== null && _b !== void 0 ? _b : "";
        return message;
    },
};
function isSet(value) {
    return value !== null && value !== undefined;
}
