// secretjs/dist/protobuf/ibc-apps/packetforward/v1/genesis.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.178.0
//   protoc               v3.21.3
// source: ibc-apps/packetforward/v1/genesis.proto
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InFlightPacket = exports.Params = exports.GenesisState_InFlightPacketsEntry = exports.GenesisState = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const _m0 = __importStar(require("protobufjs/minimal"));
exports.protobufPackage = "packetforward.v1";
function createBaseGenesisState() {
    return { params: undefined, in_flight_packets: {} };
}
exports.GenesisState = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.params !== undefined) {
            exports.Params.encode(message.params, writer.uint32(10).fork()).ldelim();
        }
        Object.entries(message.in_flight_packets).forEach(([key, value]) => {
            exports.GenesisState_InFlightPacketsEntry.encode({ key: key, value }, writer.uint32(18).fork()).ldelim();
        });
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGenesisState();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.params = exports.Params.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    const entry2 = exports.GenesisState_InFlightPacketsEntry.decode(reader, reader.uint32());
                    if (entry2.value !== undefined) {
                        message.in_flight_packets[entry2.key] = entry2.value;
                    }
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            params: isSet(object.params) ? exports.Params.fromJSON(object.params) : undefined,
            in_flight_packets: isObject(object.in_flight_packets)
                ? Object.entries(object.in_flight_packets).reduce((acc, [key, value]) => {
                    acc[key] = exports.InFlightPacket.fromJSON(value);
                    return acc;
                }, {})
                : {},
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.params !== undefined) {
            obj.params = exports.Params.toJSON(message.params);
        }
        if (message.in_flight_packets) {
            const entries = Object.entries(message.in_flight_packets);
            if (entries.length > 0) {
                obj.in_flight_packets = {};
                entries.forEach(([k, v]) => {
                    obj.in_flight_packets[k] = exports.InFlightPacket.toJSON(v);
                });
            }
        }
        return obj;
    },
    create(base) {
        return exports.GenesisState.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseGenesisState();
        message.params = (object.params !== undefined && object.params !== null)
            ? exports.Params.fromPartial(object.params)
            : undefined;
        message.in_flight_packets = Object.entries((_a = object.in_flight_packets) !== null && _a !== void 0 ? _a : {}).reduce((acc, [key, value]) => {
            if (value !== undefined) {
                acc[key] = exports.InFlightPacket.fromPartial(value);
            }
            return acc;
        }, {});
        return message;
    },
};
function createBaseGenesisState_InFlightPacketsEntry() {
    return { key: "", value: undefined };
}
exports.GenesisState_InFlightPacketsEntry = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (message.value !== undefined) {
            exports.InFlightPacket.encode(message.value, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGenesisState_InFlightPacketsEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.value = exports.InFlightPacket.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? globalThis.String(object.key) : "",
            value: isSet(object.value) ? exports.InFlightPacket.fromJSON(object.value) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key !== "") {
            obj.key = message.key;
        }
        if (message.value !== undefined) {
            obj.value = exports.InFlightPacket.toJSON(message.value);
        }
        return obj;
    },
    create(base) {
        return exports.GenesisState_InFlightPacketsEntry.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseGenesisState_InFlightPacketsEntry();
        message.key = (_a = object.key) !== null && _a !== void 0 ? _a : "";
        message.value = (object.value !== undefined && object.value !== null)
            ? exports.InFlightPacket.fromPartial(object.value)
            : undefined;
        return message;
    },
};
function createBaseParams() {
    return { fee_percentage: "" };
}
exports.Params = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.fee_percentage !== "") {
            writer.uint32(10).string(message.fee_percentage);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.fee_percentage = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { fee_percentage: isSet(object.fee_percentage) ? globalThis.String(object.fee_percentage) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.fee_percentage !== "") {
            obj.fee_percentage = message.fee_percentage;
        }
        return obj;
    },
    create(base) {
        return exports.Params.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseParams();
        message.fee_percentage = (_a = object.fee_percentage) !== null && _a !== void 0 ? _a : "";
        return message;
    },
};
function createBaseInFlightPacket() {
    return {
        original_sender_address: "",
        refund_channel_id: "",
        refund_port_id: "",
        packet_src_channel_id: "",
        packet_src_port_id: "",
        packet_timeout_timestamp: "0",
        packet_timeout_height: "",
        packet_data: new Uint8Array(0),
        refund_sequence: "0",
        retries_remaining: 0,
        timeout: "0",
        nonrefundable: false,
    };
}
exports.InFlightPacket = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.original_sender_address !== "") {
            writer.uint32(10).string(message.original_sender_address);
        }
        if (message.refund_channel_id !== "") {
            writer.uint32(18).string(message.refund_channel_id);
        }
        if (message.refund_port_id !== "") {
            writer.uint32(26).string(message.refund_port_id);
        }
        if (message.packet_src_channel_id !== "") {
            writer.uint32(34).string(message.packet_src_channel_id);
        }
        if (message.packet_src_port_id !== "") {
            writer.uint32(42).string(message.packet_src_port_id);
        }
        if (message.packet_timeout_timestamp !== "0") {
            writer.uint32(48).uint64(message.packet_timeout_timestamp);
        }
        if (message.packet_timeout_height !== "") {
            writer.uint32(58).string(message.packet_timeout_height);
        }
        if (message.packet_data.length !== 0) {
            writer.uint32(66).bytes(message.packet_data);
        }
        if (message.refund_sequence !== "0") {
            writer.uint32(72).uint64(message.refund_sequence);
        }
        if (message.retries_remaining !== 0) {
            writer.uint32(80).int32(message.retries_remaining);
        }
        if (message.timeout !== "0") {
            writer.uint32(88).uint64(message.timeout);
        }
        if (message.nonrefundable !== false) {
            writer.uint32(96).bool(message.nonrefundable);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseInFlightPacket();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.original_sender_address = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.refund_channel_id = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.refund_port_id = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.packet_src_channel_id = reader.string();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.packet_src_port_id = reader.string();
                    continue;
                case 6:
                    if (tag !== 48) {
                        break;
                    }
                    message.packet_timeout_timestamp = longToString(reader.uint64());
                    continue;
                case 7:
                    if (tag !== 58) {
                        break;
                    }
                    message.packet_timeout_height = reader.string();
                    continue;
                case 8:
                    if (tag !== 66) {
                        break;
                    }
                    message.packet_data = reader.bytes();
                    continue;
                case 9:
                    if (tag !== 72) {
                        break;
                    }
                    message.refund_sequence = longToString(reader.uint64());
                    continue;
                case 10:
                    if (tag !== 80) {
                        break;
                    }
                    message.retries_remaining = reader.int32();
                    continue;
                case 11:
                    if (tag !== 88) {
                        break;
                    }
                    message.timeout = longToString(reader.uint64());
                    continue;
                case 12:
                    if (tag !== 96) {
                        break;
                    }
                    message.nonrefundable = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            original_sender_address: isSet(object.original_sender_address)
                ? globalThis.String(object.original_sender_address)
                : "",
            refund_channel_id: isSet(object.refund_channel_id) ? globalThis.String(object.refund_channel_id) : "",
            refund_port_id: isSet(object.refund_port_id) ? globalThis.String(object.refund_port_id) : "",
            packet_src_channel_id: isSet(object.packet_src_channel_id) ? globalThis.String(object.packet_src_channel_id) : "",
            packet_src_port_id: isSet(object.packet_src_port_id) ? globalThis.String(object.packet_src_port_id) : "",
            packet_timeout_timestamp: isSet(object.packet_timeout_timestamp)
                ? globalThis.String(object.packet_timeout_timestamp)
                : "0",
            packet_timeout_height: isSet(object.packet_timeout_height) ? globalThis.String(object.packet_timeout_height) : "",
            packet_data: isSet(object.packet_data) ? bytesFromBase64(object.packet_data) : new Uint8Array(0),
            refund_sequence: isSet(object.refund_sequence) ? globalThis.String(object.refund_sequence) : "0",
            retries_remaining: isSet(object.retries_remaining) ? globalThis.Number(object.retries_remaining) : 0,
            timeout: isSet(object.timeout) ? globalThis.String(object.timeout) : "0",
            nonrefundable: isSet(object.nonrefundable) ? globalThis.Boolean(object.nonrefundable) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.original_sender_address !== "") {
            obj.original_sender_address = message.original_sender_address;
        }
        if (message.refund_channel_id !== "") {
            obj.refund_channel_id = message.refund_channel_id;
        }
        if (message.refund_port_id !== "") {
            obj.refund_port_id = message.refund_port_id;
        }
        if (message.packet_src_channel_id !== "") {
            obj.packet_src_channel_id = message.packet_src_channel_id;
        }
        if (message.packet_src_port_id !== "") {
            obj.packet_src_port_id = message.packet_src_port_id;
        }
        if (message.packet_timeout_timestamp !== "0") {
            obj.packet_timeout_timestamp = message.packet_timeout_timestamp;
        }
        if (message.packet_timeout_height !== "") {
            obj.packet_timeout_height = message.packet_timeout_height;
        }
        if (message.packet_data.length !== 0) {
            obj.packet_data = base64FromBytes(message.packet_data);
        }
        if (message.refund_sequence !== "0") {
            obj.refund_sequence = message.refund_sequence;
        }
        if (message.retries_remaining !== 0) {
            obj.retries_remaining = Math.round(message.retries_remaining);
        }
        if (message.timeout !== "0") {
            obj.timeout = message.timeout;
        }
        if (message.nonrefundable !== false) {
            obj.nonrefundable = message.nonrefundable;
        }
        return obj;
    },
    create(base) {
        return exports.InFlightPacket.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m;
        const message = createBaseInFlightPacket();
        message.original_sender_address = (_a = object.original_sender_address) !== null && _a !== void 0 ? _a : "";
        message.refund_channel_id = (_b = object.refund_channel_id) !== null && _b !== void 0 ? _b : "";
        message.refund_port_id = (_c = object.refund_port_id) !== null && _c !== void 0 ? _c : "";
        message.packet_src_channel_id = (_d = object.packet_src_channel_id) !== null && _d !== void 0 ? _d : "";
        message.packet_src_port_id = (_e = object.packet_src_port_id) !== null && _e !== void 0 ? _e : "";
        message.packet_timeout_timestamp = (_f = object.packet_timeout_timestamp) !== null && _f !== void 0 ? _f : "0";
        message.packet_timeout_height = (_g = object.packet_timeout_height) !== null && _g !== void 0 ? _g : "";
        message.packet_data = (_h = object.packet_data) !== null && _h !== void 0 ? _h : new Uint8Array(0);
        message.refund_sequence = (_j = object.refund_sequence) !== null && _j !== void 0 ? _j : "0";
        message.retries_remaining = (_k = object.retries_remaining) !== null && _k !== void 0 ? _k : 0;
        message.timeout = (_l = object.timeout) !== null && _l !== void 0 ? _l : "0";
        message.nonrefundable = (_m = object.nonrefundable) !== null && _m !== void 0 ? _m : false;
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
function longToString(long) {
    return long.toString();
}
if (_m0.util.Long !== long_1.default) {
    _m0.util.Long = long_1.default;
    _m0.configure();
}
function isObject(value) {
    return typeof value === "object" && value !== null;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
