// secretjs/dist/grpc_gateway/tendermint/abci/types.pb.js
"use strict";
/* eslint-disable */
// @ts-nocheck
/*
* This file is a generated Typescript file for GRPC Gateway, DO NOT MODIFY
*/
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ABCI = exports.ResponseVerifyVoteExtensionVerifyStatus = exports.ResponseProcessProposalProposalStatus = exports.ResponseApplySnapshotChunkResult = exports.ResponseOfferSnapshotResult = exports.MisbehaviorType = exports.CheckTxType = void 0;
const fm = __importStar(require("../../fetch.pb"));
var CheckTxType;
(function (CheckTxType) {
    CheckTxType["NEW"] = "NEW";
    CheckTxType["RECHECK"] = "RECHECK";
})(CheckTxType = exports.CheckTxType || (exports.CheckTxType = {}));
var MisbehaviorType;
(function (MisbehaviorType) {
    MisbehaviorType["UNKNOWN"] = "UNKNOWN";
    MisbehaviorType["DUPLICATE_VOTE"] = "DUPLICATE_VOTE";
    MisbehaviorType["LIGHT_CLIENT_ATTACK"] = "LIGHT_CLIENT_ATTACK";
})(MisbehaviorType = exports.MisbehaviorType || (exports.MisbehaviorType = {}));
var ResponseOfferSnapshotResult;
(function (ResponseOfferSnapshotResult) {
    ResponseOfferSnapshotResult["UNKNOWN"] = "UNKNOWN";
    ResponseOfferSnapshotResult["ACCEPT"] = "ACCEPT";
    ResponseOfferSnapshotResult["ABORT"] = "ABORT";
    ResponseOfferSnapshotResult["REJECT"] = "REJECT";
    ResponseOfferSnapshotResult["REJECT_FORMAT"] = "REJECT_FORMAT";
    ResponseOfferSnapshotResult["REJECT_SENDER"] = "REJECT_SENDER";
})(ResponseOfferSnapshotResult = exports.ResponseOfferSnapshotResult || (exports.ResponseOfferSnapshotResult = {}));
var ResponseApplySnapshotChunkResult;
(function (ResponseApplySnapshotChunkResult) {
    ResponseApplySnapshotChunkResult["UNKNOWN"] = "UNKNOWN";
    ResponseApplySnapshotChunkResult["ACCEPT"] = "ACCEPT";
    ResponseApplySnapshotChunkResult["ABORT"] = "ABORT";
    ResponseApplySnapshotChunkResult["RETRY"] = "RETRY";
    ResponseApplySnapshotChunkResult["RETRY_SNAPSHOT"] = "RETRY_SNAPSHOT";
    ResponseApplySnapshotChunkResult["REJECT_SNAPSHOT"] = "REJECT_SNAPSHOT";
})(ResponseApplySnapshotChunkResult = exports.ResponseApplySnapshotChunkResult || (exports.ResponseApplySnapshotChunkResult = {}));
var ResponseProcessProposalProposalStatus;
(function (ResponseProcessProposalProposalStatus) {
    ResponseProcessProposalProposalStatus["UNKNOWN"] = "UNKNOWN";
    ResponseProcessProposalProposalStatus["ACCEPT"] = "ACCEPT";
    ResponseProcessProposalProposalStatus["REJECT"] = "REJECT";
})(ResponseProcessProposalProposalStatus = exports.ResponseProcessProposalProposalStatus || (exports.ResponseProcessProposalProposalStatus = {}));
var ResponseVerifyVoteExtensionVerifyStatus;
(function (ResponseVerifyVoteExtensionVerifyStatus) {
    ResponseVerifyVoteExtensionVerifyStatus["UNKNOWN"] = "UNKNOWN";
    ResponseVerifyVoteExtensionVerifyStatus["ACCEPT"] = "ACCEPT";
    ResponseVerifyVoteExtensionVerifyStatus["REJECT"] = "REJECT";
})(ResponseVerifyVoteExtensionVerifyStatus = exports.ResponseVerifyVoteExtensionVerifyStatus || (exports.ResponseVerifyVoteExtensionVerifyStatus = {}));
class ABCI {
    static Echo(req, initReq) {
        return fm.fetchReq(`/tendermint.abci.ABCI/Echo`, Object.assign(Object.assign({}, initReq), { method: "POST", body: JSON.stringify(req, fm.replacer) }));
    }
    static Flush(req, initReq) {
        return fm.fetchReq(`/tendermint.abci.ABCI/Flush`, Object.assign(Object.assign({}, initReq), { method: "POST", body: JSON.stringify(req, fm.replacer) }));
    }
    static Info(req, initReq) {
        return fm.fetchReq(`/tendermint.abci.ABCI/Info`, Object.assign(Object.assign({}, initReq), { method: "POST", body: JSON.stringify(req, fm.replacer) }));
    }
    static CheckTx(req, initReq) {
        return fm.fetchReq(`/tendermint.abci.ABCI/CheckTx`, Object.assign(Object.assign({}, initReq), { method: "POST", body: JSON.stringify(req, fm.replacer) }));
    }
    static Query(req, initReq) {
        return fm.fetchReq(`/tendermint.abci.ABCI/Query`, Object.assign(Object.assign({}, initReq), { method: "POST", body: JSON.stringify(req, fm.replacer) }));
    }
    static Commit(req, initReq) {
        return fm.fetchReq(`/tendermint.abci.ABCI/Commit`, Object.assign(Object.assign({}, initReq), { method: "POST", body: JSON.stringify(req, fm.replacer) }));
    }
    static InitChain(req, initReq) {
        return fm.fetchReq(`/tendermint.abci.ABCI/InitChain`, Object.assign(Object.assign({}, initReq), { method: "POST", body: JSON.stringify(req, fm.replacer) }));
    }
    static ListSnapshots(req, initReq) {
        return fm.fetchReq(`/tendermint.abci.ABCI/ListSnapshots`, Object.assign(Object.assign({}, initReq), { method: "POST", body: JSON.stringify(req, fm.replacer) }));
    }
    static OfferSnapshot(req, initReq) {
        return fm.fetchReq(`/tendermint.abci.ABCI/OfferSnapshot`, Object.assign(Object.assign({}, initReq), { method: "POST", body: JSON.stringify(req, fm.replacer) }));
    }
    static LoadSnapshotChunk(req, initReq) {
        return fm.fetchReq(`/tendermint.abci.ABCI/LoadSnapshotChunk`, Object.assign(Object.assign({}, initReq), { method: "POST", body: JSON.stringify(req, fm.replacer) }));
    }
    static ApplySnapshotChunk(req, initReq) {
        return fm.fetchReq(`/tendermint.abci.ABCI/ApplySnapshotChunk`, Object.assign(Object.assign({}, initReq), { method: "POST", body: JSON.stringify(req, fm.replacer) }));
    }
    static PrepareProposal(req, initReq) {
        return fm.fetchReq(`/tendermint.abci.ABCI/PrepareProposal`, Object.assign(Object.assign({}, initReq), { method: "POST", body: JSON.stringify(req, fm.replacer) }));
    }
    static ProcessProposal(req, initReq) {
        return fm.fetchReq(`/tendermint.abci.ABCI/ProcessProposal`, Object.assign(Object.assign({}, initReq), { method: "POST", body: JSON.stringify(req, fm.replacer) }));
    }
    static ExtendVote(req, initReq) {
        return fm.fetchReq(`/tendermint.abci.ABCI/ExtendVote`, Object.assign(Object.assign({}, initReq), { method: "POST", body: JSON.stringify(req, fm.replacer) }));
    }
    static VerifyVoteExtension(req, initReq) {
        return fm.fetchReq(`/tendermint.abci.ABCI/VerifyVoteExtension`, Object.assign(Object.assign({}, initReq), { method: "POST", body: JSON.stringify(req, fm.replacer) }));
    }
    static FinalizeBlock(req, initReq) {
        return fm.fetchReq(`/tendermint.abci.ABCI/FinalizeBlock`, Object.assign(Object.assign({}, initReq), { method: "POST", body: JSON.stringify(req, fm.replacer) }));
    }
}
exports.ABCI = ABCI;
