// secretjs/dist/protobuf/ibc/applications/fee/v1/ack.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.178.0
//   protoc               v3.21.3
// source: ibc/applications/fee/v1/ack.proto
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncentivizedAcknowledgement = exports.protobufPackage = void 0;
/* eslint-disable */
const _m0 = __importStar(require("protobufjs/minimal"));
exports.protobufPackage = "ibc.applications.fee.v1";
function createBaseIncentivizedAcknowledgement() {
    return { app_acknowledgement: new Uint8Array(0), forward_relayer_address: "", underlying_app_success: false };
}
exports.IncentivizedAcknowledgement = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.app_acknowledgement.length !== 0) {
            writer.uint32(10).bytes(message.app_acknowledgement);
        }
        if (message.forward_relayer_address !== "") {
            writer.uint32(18).string(message.forward_relayer_address);
        }
        if (message.underlying_app_success !== false) {
            writer.uint32(24).bool(message.underlying_app_success);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseIncentivizedAcknowledgement();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.app_acknowledgement = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.forward_relayer_address = reader.string();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.underlying_app_success = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            app_acknowledgement: isSet(object.app_acknowledgement)
                ? bytesFromBase64(object.app_acknowledgement)
                : new Uint8Array(0),
            forward_relayer_address: isSet(object.forward_relayer_address)
                ? globalThis.String(object.forward_relayer_address)
                : "",
            underlying_app_success: isSet(object.underlying_app_success)
                ? globalThis.Boolean(object.underlying_app_success)
                : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.app_acknowledgement.length !== 0) {
            obj.app_acknowledgement = base64FromBytes(message.app_acknowledgement);
        }
        if (message.forward_relayer_address !== "") {
            obj.forward_relayer_address = message.forward_relayer_address;
        }
        if (message.underlying_app_success !== false) {
            obj.underlying_app_success = message.underlying_app_success;
        }
        return obj;
    },
    create(base) {
        return exports.IncentivizedAcknowledgement.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseIncentivizedAcknowledgement();
        message.app_acknowledgement = (_a = object.app_acknowledgement) !== null && _a !== void 0 ? _a : new Uint8Array(0);
        message.forward_relayer_address = (_b = object.forward_relayer_address) !== null && _b !== void 0 ? _b : "";
        message.underlying_app_success = (_c = object.underlying_app_success) !== null && _c !== void 0 ? _c : false;
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
