// secretjs/dist/protobuf/tendermint/state/types.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.178.0
//   protoc               v3.21.3
// source: tendermint/state/types.proto
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.State = exports.Version = exports.ABCIResponsesInfo = exports.ConsensusParamsInfo = exports.ValidatorsInfo = exports.ResponseEndBlock = exports.ResponseBeginBlock = exports.LegacyABCIResponses = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const _m0 = __importStar(require("protobufjs/minimal"));
const timestamp_1 = require("../../google/protobuf/timestamp");
const types_1 = require("../abci/types");
const params_1 = require("../types/params");
const types_2 = require("../types/types");
const validator_1 = require("../types/validator");
const types_3 = require("../version/types");
exports.protobufPackage = "tendermint.state";
function createBaseLegacyABCIResponses() {
    return { deliver_txs: [], end_block: undefined, begin_block: undefined };
}
exports.LegacyABCIResponses = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.deliver_txs) {
            types_1.ExecTxResult.encode(v, writer.uint32(10).fork()).ldelim();
        }
        if (message.end_block !== undefined) {
            exports.ResponseEndBlock.encode(message.end_block, writer.uint32(18).fork()).ldelim();
        }
        if (message.begin_block !== undefined) {
            exports.ResponseBeginBlock.encode(message.begin_block, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseLegacyABCIResponses();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.deliver_txs.push(types_1.ExecTxResult.decode(reader, reader.uint32()));
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.end_block = exports.ResponseEndBlock.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.begin_block = exports.ResponseBeginBlock.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            deliver_txs: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.deliver_txs)
                ? object.deliver_txs.map((e) => types_1.ExecTxResult.fromJSON(e))
                : [],
            end_block: isSet(object.end_block) ? exports.ResponseEndBlock.fromJSON(object.end_block) : undefined,
            begin_block: isSet(object.begin_block) ? exports.ResponseBeginBlock.fromJSON(object.begin_block) : undefined,
        };
    },
    toJSON(message) {
        var _a;
        const obj = {};
        if ((_a = message.deliver_txs) === null || _a === void 0 ? void 0 : _a.length) {
            obj.deliver_txs = message.deliver_txs.map((e) => types_1.ExecTxResult.toJSON(e));
        }
        if (message.end_block !== undefined) {
            obj.end_block = exports.ResponseEndBlock.toJSON(message.end_block);
        }
        if (message.begin_block !== undefined) {
            obj.begin_block = exports.ResponseBeginBlock.toJSON(message.begin_block);
        }
        return obj;
    },
    create(base) {
        return exports.LegacyABCIResponses.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseLegacyABCIResponses();
        message.deliver_txs = ((_a = object.deliver_txs) === null || _a === void 0 ? void 0 : _a.map((e) => types_1.ExecTxResult.fromPartial(e))) || [];
        message.end_block = (object.end_block !== undefined && object.end_block !== null)
            ? exports.ResponseEndBlock.fromPartial(object.end_block)
            : undefined;
        message.begin_block = (object.begin_block !== undefined && object.begin_block !== null)
            ? exports.ResponseBeginBlock.fromPartial(object.begin_block)
            : undefined;
        return message;
    },
};
function createBaseResponseBeginBlock() {
    return { events: [] };
}
exports.ResponseBeginBlock = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.events) {
            types_1.Event.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseResponseBeginBlock();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.events.push(types_1.Event.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { events: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.events) ? object.events.map((e) => types_1.Event.fromJSON(e)) : [] };
    },
    toJSON(message) {
        var _a;
        const obj = {};
        if ((_a = message.events) === null || _a === void 0 ? void 0 : _a.length) {
            obj.events = message.events.map((e) => types_1.Event.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.ResponseBeginBlock.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseResponseBeginBlock();
        message.events = ((_a = object.events) === null || _a === void 0 ? void 0 : _a.map((e) => types_1.Event.fromPartial(e))) || [];
        return message;
    },
};
function createBaseResponseEndBlock() {
    return { validator_updates: [], consensus_param_updates: undefined, events: [] };
}
exports.ResponseEndBlock = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.validator_updates) {
            types_1.ValidatorUpdate.encode(v, writer.uint32(10).fork()).ldelim();
        }
        if (message.consensus_param_updates !== undefined) {
            params_1.ConsensusParams.encode(message.consensus_param_updates, writer.uint32(18).fork()).ldelim();
        }
        for (const v of message.events) {
            types_1.Event.encode(v, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseResponseEndBlock();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.validator_updates.push(types_1.ValidatorUpdate.decode(reader, reader.uint32()));
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.consensus_param_updates = params_1.ConsensusParams.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.events.push(types_1.Event.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            validator_updates: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.validator_updates)
                ? object.validator_updates.map((e) => types_1.ValidatorUpdate.fromJSON(e))
                : [],
            consensus_param_updates: isSet(object.consensus_param_updates)
                ? params_1.ConsensusParams.fromJSON(object.consensus_param_updates)
                : undefined,
            events: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.events) ? object.events.map((e) => types_1.Event.fromJSON(e)) : [],
        };
    },
    toJSON(message) {
        var _a, _b;
        const obj = {};
        if ((_a = message.validator_updates) === null || _a === void 0 ? void 0 : _a.length) {
            obj.validator_updates = message.validator_updates.map((e) => types_1.ValidatorUpdate.toJSON(e));
        }
        if (message.consensus_param_updates !== undefined) {
            obj.consensus_param_updates = params_1.ConsensusParams.toJSON(message.consensus_param_updates);
        }
        if ((_b = message.events) === null || _b === void 0 ? void 0 : _b.length) {
            obj.events = message.events.map((e) => types_1.Event.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.ResponseEndBlock.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseResponseEndBlock();
        message.validator_updates = ((_a = object.validator_updates) === null || _a === void 0 ? void 0 : _a.map((e) => types_1.ValidatorUpdate.fromPartial(e))) || [];
        message.consensus_param_updates =
            (object.consensus_param_updates !== undefined && object.consensus_param_updates !== null)
                ? params_1.ConsensusParams.fromPartial(object.consensus_param_updates)
                : undefined;
        message.events = ((_b = object.events) === null || _b === void 0 ? void 0 : _b.map((e) => types_1.Event.fromPartial(e))) || [];
        return message;
    },
};
function createBaseValidatorsInfo() {
    return { validator_set: undefined, last_height_changed: "0" };
}
exports.ValidatorsInfo = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.validator_set !== undefined) {
            validator_1.ValidatorSet.encode(message.validator_set, writer.uint32(10).fork()).ldelim();
        }
        if (message.last_height_changed !== "0") {
            writer.uint32(16).int64(message.last_height_changed);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidatorsInfo();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.validator_set = validator_1.ValidatorSet.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.last_height_changed = longToString(reader.int64());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            validator_set: isSet(object.validator_set) ? validator_1.ValidatorSet.fromJSON(object.validator_set) : undefined,
            last_height_changed: isSet(object.last_height_changed) ? globalThis.String(object.last_height_changed) : "0",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.validator_set !== undefined) {
            obj.validator_set = validator_1.ValidatorSet.toJSON(message.validator_set);
        }
        if (message.last_height_changed !== "0") {
            obj.last_height_changed = message.last_height_changed;
        }
        return obj;
    },
    create(base) {
        return exports.ValidatorsInfo.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseValidatorsInfo();
        message.validator_set = (object.validator_set !== undefined && object.validator_set !== null)
            ? validator_1.ValidatorSet.fromPartial(object.validator_set)
            : undefined;
        message.last_height_changed = (_a = object.last_height_changed) !== null && _a !== void 0 ? _a : "0";
        return message;
    },
};
function createBaseConsensusParamsInfo() {
    return { consensus_params: undefined, last_height_changed: "0" };
}
exports.ConsensusParamsInfo = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.consensus_params !== undefined) {
            params_1.ConsensusParams.encode(message.consensus_params, writer.uint32(10).fork()).ldelim();
        }
        if (message.last_height_changed !== "0") {
            writer.uint32(16).int64(message.last_height_changed);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConsensusParamsInfo();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.consensus_params = params_1.ConsensusParams.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.last_height_changed = longToString(reader.int64());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            consensus_params: isSet(object.consensus_params) ? params_1.ConsensusParams.fromJSON(object.consensus_params) : undefined,
            last_height_changed: isSet(object.last_height_changed) ? globalThis.String(object.last_height_changed) : "0",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.consensus_params !== undefined) {
            obj.consensus_params = params_1.ConsensusParams.toJSON(message.consensus_params);
        }
        if (message.last_height_changed !== "0") {
            obj.last_height_changed = message.last_height_changed;
        }
        return obj;
    },
    create(base) {
        return exports.ConsensusParamsInfo.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseConsensusParamsInfo();
        message.consensus_params = (object.consensus_params !== undefined && object.consensus_params !== null)
            ? params_1.ConsensusParams.fromPartial(object.consensus_params)
            : undefined;
        message.last_height_changed = (_a = object.last_height_changed) !== null && _a !== void 0 ? _a : "0";
        return message;
    },
};
function createBaseABCIResponsesInfo() {
    return { legacy_abci_responses: undefined, height: "0", response_finalize_block: undefined };
}
exports.ABCIResponsesInfo = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.legacy_abci_responses !== undefined) {
            exports.LegacyABCIResponses.encode(message.legacy_abci_responses, writer.uint32(10).fork()).ldelim();
        }
        if (message.height !== "0") {
            writer.uint32(16).int64(message.height);
        }
        if (message.response_finalize_block !== undefined) {
            types_1.ResponseFinalizeBlock.encode(message.response_finalize_block, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseABCIResponsesInfo();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.legacy_abci_responses = exports.LegacyABCIResponses.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.height = longToString(reader.int64());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.response_finalize_block = types_1.ResponseFinalizeBlock.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            legacy_abci_responses: isSet(object.legacy_abci_responses)
                ? exports.LegacyABCIResponses.fromJSON(object.legacy_abci_responses)
                : undefined,
            height: isSet(object.height) ? globalThis.String(object.height) : "0",
            response_finalize_block: isSet(object.response_finalize_block)
                ? types_1.ResponseFinalizeBlock.fromJSON(object.response_finalize_block)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.legacy_abci_responses !== undefined) {
            obj.legacy_abci_responses = exports.LegacyABCIResponses.toJSON(message.legacy_abci_responses);
        }
        if (message.height !== "0") {
            obj.height = message.height;
        }
        if (message.response_finalize_block !== undefined) {
            obj.response_finalize_block = types_1.ResponseFinalizeBlock.toJSON(message.response_finalize_block);
        }
        return obj;
    },
    create(base) {
        return exports.ABCIResponsesInfo.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseABCIResponsesInfo();
        message.legacy_abci_responses =
            (object.legacy_abci_responses !== undefined && object.legacy_abci_responses !== null)
                ? exports.LegacyABCIResponses.fromPartial(object.legacy_abci_responses)
                : undefined;
        message.height = (_a = object.height) !== null && _a !== void 0 ? _a : "0";
        message.response_finalize_block =
            (object.response_finalize_block !== undefined && object.response_finalize_block !== null)
                ? types_1.ResponseFinalizeBlock.fromPartial(object.response_finalize_block)
                : undefined;
        return message;
    },
};
function createBaseVersion() {
    return { consensus: undefined, software: "" };
}
exports.Version = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.consensus !== undefined) {
            types_3.Consensus.encode(message.consensus, writer.uint32(10).fork()).ldelim();
        }
        if (message.software !== "") {
            writer.uint32(18).string(message.software);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseVersion();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.consensus = types_3.Consensus.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.software = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            consensus: isSet(object.consensus) ? types_3.Consensus.fromJSON(object.consensus) : undefined,
            software: isSet(object.software) ? globalThis.String(object.software) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.consensus !== undefined) {
            obj.consensus = types_3.Consensus.toJSON(message.consensus);
        }
        if (message.software !== "") {
            obj.software = message.software;
        }
        return obj;
    },
    create(base) {
        return exports.Version.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseVersion();
        message.consensus = (object.consensus !== undefined && object.consensus !== null)
            ? types_3.Consensus.fromPartial(object.consensus)
            : undefined;
        message.software = (_a = object.software) !== null && _a !== void 0 ? _a : "";
        return message;
    },
};
function createBaseState() {
    return {
        version: undefined,
        chain_id: "",
        initial_height: "0",
        last_block_height: "0",
        last_block_id: undefined,
        last_block_time: undefined,
        next_validators: undefined,
        validators: undefined,
        last_validators: undefined,
        last_height_validators_changed: "0",
        consensus_params: undefined,
        last_height_consensus_params_changed: "0",
        last_results_hash: new Uint8Array(0),
        app_hash: new Uint8Array(0),
    };
}
exports.State = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.version !== undefined) {
            exports.Version.encode(message.version, writer.uint32(10).fork()).ldelim();
        }
        if (message.chain_id !== "") {
            writer.uint32(18).string(message.chain_id);
        }
        if (message.initial_height !== "0") {
            writer.uint32(112).int64(message.initial_height);
        }
        if (message.last_block_height !== "0") {
            writer.uint32(24).int64(message.last_block_height);
        }
        if (message.last_block_id !== undefined) {
            types_2.BlockID.encode(message.last_block_id, writer.uint32(34).fork()).ldelim();
        }
        if (message.last_block_time !== undefined) {
            timestamp_1.Timestamp.encode(message.last_block_time, writer.uint32(42).fork()).ldelim();
        }
        if (message.next_validators !== undefined) {
            validator_1.ValidatorSet.encode(message.next_validators, writer.uint32(50).fork()).ldelim();
        }
        if (message.validators !== undefined) {
            validator_1.ValidatorSet.encode(message.validators, writer.uint32(58).fork()).ldelim();
        }
        if (message.last_validators !== undefined) {
            validator_1.ValidatorSet.encode(message.last_validators, writer.uint32(66).fork()).ldelim();
        }
        if (message.last_height_validators_changed !== "0") {
            writer.uint32(72).int64(message.last_height_validators_changed);
        }
        if (message.consensus_params !== undefined) {
            params_1.ConsensusParams.encode(message.consensus_params, writer.uint32(82).fork()).ldelim();
        }
        if (message.last_height_consensus_params_changed !== "0") {
            writer.uint32(88).int64(message.last_height_consensus_params_changed);
        }
        if (message.last_results_hash.length !== 0) {
            writer.uint32(98).bytes(message.last_results_hash);
        }
        if (message.app_hash.length !== 0) {
            writer.uint32(106).bytes(message.app_hash);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseState();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.version = exports.Version.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.chain_id = reader.string();
                    continue;
                case 14:
                    if (tag !== 112) {
                        break;
                    }
                    message.initial_height = longToString(reader.int64());
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.last_block_height = longToString(reader.int64());
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.last_block_id = types_2.BlockID.decode(reader, reader.uint32());
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.last_block_time = timestamp_1.Timestamp.decode(reader, reader.uint32());
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.next_validators = validator_1.ValidatorSet.decode(reader, reader.uint32());
                    continue;
                case 7:
                    if (tag !== 58) {
                        break;
                    }
                    message.validators = validator_1.ValidatorSet.decode(reader, reader.uint32());
                    continue;
                case 8:
                    if (tag !== 66) {
                        break;
                    }
                    message.last_validators = validator_1.ValidatorSet.decode(reader, reader.uint32());
                    continue;
                case 9:
                    if (tag !== 72) {
                        break;
                    }
                    message.last_height_validators_changed = longToString(reader.int64());
                    continue;
                case 10:
                    if (tag !== 82) {
                        break;
                    }
                    message.consensus_params = params_1.ConsensusParams.decode(reader, reader.uint32());
                    continue;
                case 11:
                    if (tag !== 88) {
                        break;
                    }
                    message.last_height_consensus_params_changed = longToString(reader.int64());
                    continue;
                case 12:
                    if (tag !== 98) {
                        break;
                    }
                    message.last_results_hash = reader.bytes();
                    continue;
                case 13:
                    if (tag !== 106) {
                        break;
                    }
                    message.app_hash = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            version: isSet(object.version) ? exports.Version.fromJSON(object.version) : undefined,
            chain_id: isSet(object.chain_id) ? globalThis.String(object.chain_id) : "",
            initial_height: isSet(object.initial_height) ? globalThis.String(object.initial_height) : "0",
            last_block_height: isSet(object.last_block_height) ? globalThis.String(object.last_block_height) : "0",
            last_block_id: isSet(object.last_block_id) ? types_2.BlockID.fromJSON(object.last_block_id) : undefined,
            last_block_time: isSet(object.last_block_time) ? fromJsonTimestamp(object.last_block_time) : undefined,
            next_validators: isSet(object.next_validators) ? validator_1.ValidatorSet.fromJSON(object.next_validators) : undefined,
            validators: isSet(object.validators) ? validator_1.ValidatorSet.fromJSON(object.validators) : undefined,
            last_validators: isSet(object.last_validators) ? validator_1.ValidatorSet.fromJSON(object.last_validators) : undefined,
            last_height_validators_changed: isSet(object.last_height_validators_changed)
                ? globalThis.String(object.last_height_validators_changed)
                : "0",
            consensus_params: isSet(object.consensus_params) ? params_1.ConsensusParams.fromJSON(object.consensus_params) : undefined,
            last_height_consensus_params_changed: isSet(object.last_height_consensus_params_changed)
                ? globalThis.String(object.last_height_consensus_params_changed)
                : "0",
            last_results_hash: isSet(object.last_results_hash)
                ? bytesFromBase64(object.last_results_hash)
                : new Uint8Array(0),
            app_hash: isSet(object.app_hash) ? bytesFromBase64(object.app_hash) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.version !== undefined) {
            obj.version = exports.Version.toJSON(message.version);
        }
        if (message.chain_id !== "") {
            obj.chain_id = message.chain_id;
        }
        if (message.initial_height !== "0") {
            obj.initial_height = message.initial_height;
        }
        if (message.last_block_height !== "0") {
            obj.last_block_height = message.last_block_height;
        }
        if (message.last_block_id !== undefined) {
            obj.last_block_id = types_2.BlockID.toJSON(message.last_block_id);
        }
        if (message.last_block_time !== undefined) {
            obj.last_block_time = fromTimestamp(message.last_block_time).toISOString();
        }
        if (message.next_validators !== undefined) {
            obj.next_validators = validator_1.ValidatorSet.toJSON(message.next_validators);
        }
        if (message.validators !== undefined) {
            obj.validators = validator_1.ValidatorSet.toJSON(message.validators);
        }
        if (message.last_validators !== undefined) {
            obj.last_validators = validator_1.ValidatorSet.toJSON(message.last_validators);
        }
        if (message.last_height_validators_changed !== "0") {
            obj.last_height_validators_changed = message.last_height_validators_changed;
        }
        if (message.consensus_params !== undefined) {
            obj.consensus_params = params_1.ConsensusParams.toJSON(message.consensus_params);
        }
        if (message.last_height_consensus_params_changed !== "0") {
            obj.last_height_consensus_params_changed = message.last_height_consensus_params_changed;
        }
        if (message.last_results_hash.length !== 0) {
            obj.last_results_hash = base64FromBytes(message.last_results_hash);
        }
        if (message.app_hash.length !== 0) {
            obj.app_hash = base64FromBytes(message.app_hash);
        }
        return obj;
    },
    create(base) {
        return exports.State.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e, _f, _g;
        const message = createBaseState();
        message.version = (object.version !== undefined && object.version !== null)
            ? exports.Version.fromPartial(object.version)
            : undefined;
        message.chain_id = (_a = object.chain_id) !== null && _a !== void 0 ? _a : "";
        message.initial_height = (_b = object.initial_height) !== null && _b !== void 0 ? _b : "0";
        message.last_block_height = (_c = object.last_block_height) !== null && _c !== void 0 ? _c : "0";
        message.last_block_id = (object.last_block_id !== undefined && object.last_block_id !== null)
            ? types_2.BlockID.fromPartial(object.last_block_id)
            : undefined;
        message.last_block_time = (object.last_block_time !== undefined && object.last_block_time !== null)
            ? timestamp_1.Timestamp.fromPartial(object.last_block_time)
            : undefined;
        message.next_validators = (object.next_validators !== undefined && object.next_validators !== null)
            ? validator_1.ValidatorSet.fromPartial(object.next_validators)
            : undefined;
        message.validators = (object.validators !== undefined && object.validators !== null)
            ? validator_1.ValidatorSet.fromPartial(object.validators)
            : undefined;
        message.last_validators = (object.last_validators !== undefined && object.last_validators !== null)
            ? validator_1.ValidatorSet.fromPartial(object.last_validators)
            : undefined;
        message.last_height_validators_changed = (_d = object.last_height_validators_changed) !== null && _d !== void 0 ? _d : "0";
        message.consensus_params = (object.consensus_params !== undefined && object.consensus_params !== null)
            ? params_1.ConsensusParams.fromPartial(object.consensus_params)
            : undefined;
        message.last_height_consensus_params_changed = (_e = object.last_height_consensus_params_changed) !== null && _e !== void 0 ? _e : "0";
        message.last_results_hash = (_f = object.last_results_hash) !== null && _f !== void 0 ? _f : new Uint8Array(0);
        message.app_hash = (_g = object.app_hash) !== null && _g !== void 0 ? _g : new Uint8Array(0);
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
function toTimestamp(date) {
    const seconds = Math.trunc(date.getTime() / 1000).toString();
    const nanos = (date.getTime() % 1000) * 1000000;
    return { seconds, nanos };
}
function fromTimestamp(t) {
    let millis = (globalThis.Number(t.seconds) || 0) * 1000;
    millis += (t.nanos || 0) / 1000000;
    return new globalThis.Date(millis);
}
function fromJsonTimestamp(o) {
    if (o instanceof globalThis.Date) {
        return toTimestamp(o);
    }
    else if (typeof o === "string") {
        return toTimestamp(new globalThis.Date(o));
    }
    else {
        return timestamp_1.Timestamp.fromJSON(o);
    }
}
function longToString(long) {
    return long.toString();
}
if (_m0.util.Long !== long_1.default) {
    _m0.util.Long = long_1.default;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
