// secretjs/dist/protobuf/cosmos/staking/module/v1/module.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.178.0
//   protoc               v3.21.3
// source: cosmos/staking/module/v1/module.proto
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Module = exports.protobufPackage = void 0;
/* eslint-disable */
const _m0 = __importStar(require("protobufjs/minimal"));
exports.protobufPackage = "cosmos.staking.module.v1";
function createBaseModule() {
    return { hooks_order: [], authority: "", bech32_prefix_validator: "", bech32_prefix_consensus: "" };
}
exports.Module = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.hooks_order) {
            writer.uint32(10).string(v);
        }
        if (message.authority !== "") {
            writer.uint32(18).string(message.authority);
        }
        if (message.bech32_prefix_validator !== "") {
            writer.uint32(26).string(message.bech32_prefix_validator);
        }
        if (message.bech32_prefix_consensus !== "") {
            writer.uint32(34).string(message.bech32_prefix_consensus);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseModule();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.hooks_order.push(reader.string());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.authority = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.bech32_prefix_validator = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.bech32_prefix_consensus = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            hooks_order: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.hooks_order)
                ? object.hooks_order.map((e) => globalThis.String(e))
                : [],
            authority: isSet(object.authority) ? globalThis.String(object.authority) : "",
            bech32_prefix_validator: isSet(object.bech32_prefix_validator)
                ? globalThis.String(object.bech32_prefix_validator)
                : "",
            bech32_prefix_consensus: isSet(object.bech32_prefix_consensus)
                ? globalThis.String(object.bech32_prefix_consensus)
                : "",
        };
    },
    toJSON(message) {
        var _a;
        const obj = {};
        if ((_a = message.hooks_order) === null || _a === void 0 ? void 0 : _a.length) {
            obj.hooks_order = message.hooks_order;
        }
        if (message.authority !== "") {
            obj.authority = message.authority;
        }
        if (message.bech32_prefix_validator !== "") {
            obj.bech32_prefix_validator = message.bech32_prefix_validator;
        }
        if (message.bech32_prefix_consensus !== "") {
            obj.bech32_prefix_consensus = message.bech32_prefix_consensus;
        }
        return obj;
    },
    create(base) {
        return exports.Module.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d;
        const message = createBaseModule();
        message.hooks_order = ((_a = object.hooks_order) === null || _a === void 0 ? void 0 : _a.map((e) => e)) || [];
        message.authority = (_b = object.authority) !== null && _b !== void 0 ? _b : "";
        message.bech32_prefix_validator = (_c = object.bech32_prefix_validator) !== null && _c !== void 0 ? _c : "";
        message.bech32_prefix_consensus = (_d = object.bech32_prefix_consensus) !== null && _d !== void 0 ? _d : "";
        return message;
    },
};
function isSet(value) {
    return value !== null && value !== undefined;
}
