// secretjs/dist/protobuf/cosmos/tx/config/v1/config.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.178.0
//   protoc               v3.21.3
// source: cosmos/tx/config/v1/config.proto
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Config = exports.protobufPackage = void 0;
/* eslint-disable */
const _m0 = __importStar(require("protobufjs/minimal"));
exports.protobufPackage = "cosmos.tx.config.v1";
function createBaseConfig() {
    return { skip_ante_handler: false, skip_post_handler: false };
}
exports.Config = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.skip_ante_handler !== false) {
            writer.uint32(8).bool(message.skip_ante_handler);
        }
        if (message.skip_post_handler !== false) {
            writer.uint32(16).bool(message.skip_post_handler);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConfig();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.skip_ante_handler = reader.bool();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.skip_post_handler = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            skip_ante_handler: isSet(object.skip_ante_handler) ? globalThis.Boolean(object.skip_ante_handler) : false,
            skip_post_handler: isSet(object.skip_post_handler) ? globalThis.Boolean(object.skip_post_handler) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.skip_ante_handler !== false) {
            obj.skip_ante_handler = message.skip_ante_handler;
        }
        if (message.skip_post_handler !== false) {
            obj.skip_post_handler = message.skip_post_handler;
        }
        return obj;
    },
    create(base) {
        return exports.Config.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseConfig();
        message.skip_ante_handler = (_a = object.skip_ante_handler) !== null && _a !== void 0 ? _a : false;
        message.skip_post_handler = (_b = object.skip_post_handler) !== null && _b !== void 0 ? _b : false;
        return message;
    },
};
function isSet(value) {
    return value !== null && value !== undefined;
}
