// secretjs/dist/protobuf/secret/registration/v1beta1/query.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.178.0
//   protoc               v3.21.3
// source: secret/registration/v1beta1/query.proto
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryClientImpl = exports.QueryServiceName = exports.QueryEncryptedSeedResponse = exports.QueryEncryptedSeedRequest = exports.protobufPackage = void 0;
/* eslint-disable */
const _m0 = __importStar(require("protobufjs/minimal"));
const empty_1 = require("../../../google/protobuf/empty");
const msg_1 = require("./msg");
exports.protobufPackage = "secret.registration.v1beta1";
function createBaseQueryEncryptedSeedRequest() {
    return { pub_key: new Uint8Array(0) };
}
exports.QueryEncryptedSeedRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.pub_key.length !== 0) {
            writer.uint32(10).bytes(message.pub_key);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryEncryptedSeedRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.pub_key = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { pub_key: isSet(object.pub_key) ? bytesFromBase64(object.pub_key) : new Uint8Array(0) };
    },
    toJSON(message) {
        const obj = {};
        if (message.pub_key.length !== 0) {
            obj.pub_key = base64FromBytes(message.pub_key);
        }
        return obj;
    },
    create(base) {
        return exports.QueryEncryptedSeedRequest.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseQueryEncryptedSeedRequest();
        message.pub_key = (_a = object.pub_key) !== null && _a !== void 0 ? _a : new Uint8Array(0);
        return message;
    },
};
function createBaseQueryEncryptedSeedResponse() {
    return { encrypted_seed: new Uint8Array(0) };
}
exports.QueryEncryptedSeedResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.encrypted_seed.length !== 0) {
            writer.uint32(10).bytes(message.encrypted_seed);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryEncryptedSeedResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.encrypted_seed = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            encrypted_seed: isSet(object.encrypted_seed) ? bytesFromBase64(object.encrypted_seed) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.encrypted_seed.length !== 0) {
            obj.encrypted_seed = base64FromBytes(message.encrypted_seed);
        }
        return obj;
    },
    create(base) {
        return exports.QueryEncryptedSeedResponse.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseQueryEncryptedSeedResponse();
        message.encrypted_seed = (_a = object.encrypted_seed) !== null && _a !== void 0 ? _a : new Uint8Array(0);
        return message;
    },
};
exports.QueryServiceName = "secret.registration.v1beta1.Query";
class QueryClientImpl {
    constructor(rpc, opts) {
        this.service = (opts === null || opts === void 0 ? void 0 : opts.service) || exports.QueryServiceName;
        this.rpc = rpc;
        this.TxKey = this.TxKey.bind(this);
        this.RegistrationKey = this.RegistrationKey.bind(this);
        this.EncryptedSeed = this.EncryptedSeed.bind(this);
    }
    TxKey(request) {
        const data = empty_1.Empty.encode(request).finish();
        const promise = this.rpc.request(this.service, "TxKey", data);
        return promise.then((data) => msg_1.Key.decode(_m0.Reader.create(data)));
    }
    RegistrationKey(request) {
        const data = empty_1.Empty.encode(request).finish();
        const promise = this.rpc.request(this.service, "RegistrationKey", data);
        return promise.then((data) => msg_1.Key.decode(_m0.Reader.create(data)));
    }
    EncryptedSeed(request) {
        const data = exports.QueryEncryptedSeedRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "EncryptedSeed", data);
        return promise.then((data) => exports.QueryEncryptedSeedResponse.decode(_m0.Reader.create(data)));
    }
}
exports.QueryClientImpl = QueryClientImpl;
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
