// secretjs/dist/protobuf/secret/registration/v1beta1/types.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.178.0
//   protoc               v3.21.3
// source: secret/registration/v1beta1/types.proto
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegistrationNodeInfo = exports.LegacySeedConfig = exports.SeedConfig = exports.protobufPackage = void 0;
/* eslint-disable */
const _m0 = __importStar(require("protobufjs/minimal"));
exports.protobufPackage = "secret.registration.v1beta1";
function createBaseSeedConfig() {
    return { master_key: "", encrypted_key: "", version: 0 };
}
exports.SeedConfig = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.master_key !== "") {
            writer.uint32(10).string(message.master_key);
        }
        if (message.encrypted_key !== "") {
            writer.uint32(18).string(message.encrypted_key);
        }
        if (message.version !== 0) {
            writer.uint32(24).uint32(message.version);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSeedConfig();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.master_key = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.encrypted_key = reader.string();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.version = reader.uint32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            master_key: isSet(object.master_key) ? globalThis.String(object.master_key) : "",
            encrypted_key: isSet(object.encrypted_key) ? globalThis.String(object.encrypted_key) : "",
            version: isSet(object.version) ? globalThis.Number(object.version) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.master_key !== "") {
            obj.master_key = message.master_key;
        }
        if (message.encrypted_key !== "") {
            obj.encrypted_key = message.encrypted_key;
        }
        if (message.version !== 0) {
            obj.version = Math.round(message.version);
        }
        return obj;
    },
    create(base) {
        return exports.SeedConfig.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseSeedConfig();
        message.master_key = (_a = object.master_key) !== null && _a !== void 0 ? _a : "";
        message.encrypted_key = (_b = object.encrypted_key) !== null && _b !== void 0 ? _b : "";
        message.version = (_c = object.version) !== null && _c !== void 0 ? _c : 0;
        return message;
    },
};
function createBaseLegacySeedConfig() {
    return { master_cert: "", encrypted_key: "" };
}
exports.LegacySeedConfig = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.master_cert !== "") {
            writer.uint32(10).string(message.master_cert);
        }
        if (message.encrypted_key !== "") {
            writer.uint32(18).string(message.encrypted_key);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseLegacySeedConfig();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.master_cert = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.encrypted_key = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            master_cert: isSet(object.master_cert) ? globalThis.String(object.master_cert) : "",
            encrypted_key: isSet(object.encrypted_key) ? globalThis.String(object.encrypted_key) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.master_cert !== "") {
            obj.master_cert = message.master_cert;
        }
        if (message.encrypted_key !== "") {
            obj.encrypted_key = message.encrypted_key;
        }
        return obj;
    },
    create(base) {
        return exports.LegacySeedConfig.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseLegacySeedConfig();
        message.master_cert = (_a = object.master_cert) !== null && _a !== void 0 ? _a : "";
        message.encrypted_key = (_b = object.encrypted_key) !== null && _b !== void 0 ? _b : "";
        return message;
    },
};
function createBaseRegistrationNodeInfo() {
    return { certificate: new Uint8Array(0), encrypted_seed: new Uint8Array(0) };
}
exports.RegistrationNodeInfo = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.certificate.length !== 0) {
            writer.uint32(10).bytes(message.certificate);
        }
        if (message.encrypted_seed.length !== 0) {
            writer.uint32(18).bytes(message.encrypted_seed);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegistrationNodeInfo();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.certificate = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.encrypted_seed = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            certificate: isSet(object.certificate) ? bytesFromBase64(object.certificate) : new Uint8Array(0),
            encrypted_seed: isSet(object.encrypted_seed) ? bytesFromBase64(object.encrypted_seed) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.certificate.length !== 0) {
            obj.certificate = base64FromBytes(message.certificate);
        }
        if (message.encrypted_seed.length !== 0) {
            obj.encrypted_seed = base64FromBytes(message.encrypted_seed);
        }
        return obj;
    },
    create(base) {
        return exports.RegistrationNodeInfo.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseRegistrationNodeInfo();
        message.certificate = (_a = object.certificate) !== null && _a !== void 0 ? _a : new Uint8Array(0);
        message.encrypted_seed = (_b = object.encrypted_seed) !== null && _b !== void 0 ? _b : new Uint8Array(0);
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
