// secretjs/dist/protobuf/cosmos/autocli/v1/options.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.178.0
//   protoc               v3.21.3
// source: cosmos/autocli/v1/options.proto
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PositionalArgDescriptor = exports.FlagOptions = exports.RpcCommandOptions_FlagOptionsEntry = exports.RpcCommandOptions = exports.ServiceCommandDescriptor_SubCommandsEntry = exports.ServiceCommandDescriptor = exports.ModuleOptions = exports.protobufPackage = void 0;
/* eslint-disable */
const _m0 = __importStar(require("protobufjs/minimal"));
exports.protobufPackage = "cosmos.autocli.v1";
function createBaseModuleOptions() {
    return { tx: undefined, query: undefined };
}
exports.ModuleOptions = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.tx !== undefined) {
            exports.ServiceCommandDescriptor.encode(message.tx, writer.uint32(10).fork()).ldelim();
        }
        if (message.query !== undefined) {
            exports.ServiceCommandDescriptor.encode(message.query, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseModuleOptions();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.tx = exports.ServiceCommandDescriptor.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.query = exports.ServiceCommandDescriptor.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            tx: isSet(object.tx) ? exports.ServiceCommandDescriptor.fromJSON(object.tx) : undefined,
            query: isSet(object.query) ? exports.ServiceCommandDescriptor.fromJSON(object.query) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.tx !== undefined) {
            obj.tx = exports.ServiceCommandDescriptor.toJSON(message.tx);
        }
        if (message.query !== undefined) {
            obj.query = exports.ServiceCommandDescriptor.toJSON(message.query);
        }
        return obj;
    },
    create(base) {
        return exports.ModuleOptions.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        const message = createBaseModuleOptions();
        message.tx = (object.tx !== undefined && object.tx !== null)
            ? exports.ServiceCommandDescriptor.fromPartial(object.tx)
            : undefined;
        message.query = (object.query !== undefined && object.query !== null)
            ? exports.ServiceCommandDescriptor.fromPartial(object.query)
            : undefined;
        return message;
    },
};
function createBaseServiceCommandDescriptor() {
    return { service: "", rpc_command_options: [], sub_commands: {} };
}
exports.ServiceCommandDescriptor = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.service !== "") {
            writer.uint32(10).string(message.service);
        }
        for (const v of message.rpc_command_options) {
            exports.RpcCommandOptions.encode(v, writer.uint32(18).fork()).ldelim();
        }
        Object.entries(message.sub_commands).forEach(([key, value]) => {
            exports.ServiceCommandDescriptor_SubCommandsEntry.encode({ key: key, value }, writer.uint32(26).fork()).ldelim();
        });
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseServiceCommandDescriptor();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.service = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.rpc_command_options.push(exports.RpcCommandOptions.decode(reader, reader.uint32()));
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    const entry3 = exports.ServiceCommandDescriptor_SubCommandsEntry.decode(reader, reader.uint32());
                    if (entry3.value !== undefined) {
                        message.sub_commands[entry3.key] = entry3.value;
                    }
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            service: isSet(object.service) ? globalThis.String(object.service) : "",
            rpc_command_options: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.rpc_command_options)
                ? object.rpc_command_options.map((e) => exports.RpcCommandOptions.fromJSON(e))
                : [],
            sub_commands: isObject(object.sub_commands)
                ? Object.entries(object.sub_commands).reduce((acc, [key, value]) => {
                    acc[key] = exports.ServiceCommandDescriptor.fromJSON(value);
                    return acc;
                }, {})
                : {},
        };
    },
    toJSON(message) {
        var _a;
        const obj = {};
        if (message.service !== "") {
            obj.service = message.service;
        }
        if ((_a = message.rpc_command_options) === null || _a === void 0 ? void 0 : _a.length) {
            obj.rpc_command_options = message.rpc_command_options.map((e) => exports.RpcCommandOptions.toJSON(e));
        }
        if (message.sub_commands) {
            const entries = Object.entries(message.sub_commands);
            if (entries.length > 0) {
                obj.sub_commands = {};
                entries.forEach(([k, v]) => {
                    obj.sub_commands[k] = exports.ServiceCommandDescriptor.toJSON(v);
                });
            }
        }
        return obj;
    },
    create(base) {
        return exports.ServiceCommandDescriptor.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseServiceCommandDescriptor();
        message.service = (_a = object.service) !== null && _a !== void 0 ? _a : "";
        message.rpc_command_options = ((_b = object.rpc_command_options) === null || _b === void 0 ? void 0 : _b.map((e) => exports.RpcCommandOptions.fromPartial(e))) || [];
        message.sub_commands = Object.entries((_c = object.sub_commands) !== null && _c !== void 0 ? _c : {}).reduce((acc, [key, value]) => {
            if (value !== undefined) {
                acc[key] = exports.ServiceCommandDescriptor.fromPartial(value);
            }
            return acc;
        }, {});
        return message;
    },
};
function createBaseServiceCommandDescriptor_SubCommandsEntry() {
    return { key: "", value: undefined };
}
exports.ServiceCommandDescriptor_SubCommandsEntry = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (message.value !== undefined) {
            exports.ServiceCommandDescriptor.encode(message.value, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseServiceCommandDescriptor_SubCommandsEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.value = exports.ServiceCommandDescriptor.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? globalThis.String(object.key) : "",
            value: isSet(object.value) ? exports.ServiceCommandDescriptor.fromJSON(object.value) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key !== "") {
            obj.key = message.key;
        }
        if (message.value !== undefined) {
            obj.value = exports.ServiceCommandDescriptor.toJSON(message.value);
        }
        return obj;
    },
    create(base) {
        return exports.ServiceCommandDescriptor_SubCommandsEntry.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseServiceCommandDescriptor_SubCommandsEntry();
        message.key = (_a = object.key) !== null && _a !== void 0 ? _a : "";
        message.value = (object.value !== undefined && object.value !== null)
            ? exports.ServiceCommandDescriptor.fromPartial(object.value)
            : undefined;
        return message;
    },
};
function createBaseRpcCommandOptions() {
    return {
        rpc_method: "",
        use: "",
        long: "",
        short: "",
        example: "",
        alias: [],
        suggest_for: [],
        deprecated: "",
        version: "",
        flag_options: {},
        positional_args: [],
        skip: false,
    };
}
exports.RpcCommandOptions = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.rpc_method !== "") {
            writer.uint32(10).string(message.rpc_method);
        }
        if (message.use !== "") {
            writer.uint32(18).string(message.use);
        }
        if (message.long !== "") {
            writer.uint32(26).string(message.long);
        }
        if (message.short !== "") {
            writer.uint32(34).string(message.short);
        }
        if (message.example !== "") {
            writer.uint32(42).string(message.example);
        }
        for (const v of message.alias) {
            writer.uint32(50).string(v);
        }
        for (const v of message.suggest_for) {
            writer.uint32(58).string(v);
        }
        if (message.deprecated !== "") {
            writer.uint32(66).string(message.deprecated);
        }
        if (message.version !== "") {
            writer.uint32(74).string(message.version);
        }
        Object.entries(message.flag_options).forEach(([key, value]) => {
            exports.RpcCommandOptions_FlagOptionsEntry.encode({ key: key, value }, writer.uint32(82).fork()).ldelim();
        });
        for (const v of message.positional_args) {
            exports.PositionalArgDescriptor.encode(v, writer.uint32(90).fork()).ldelim();
        }
        if (message.skip !== false) {
            writer.uint32(96).bool(message.skip);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRpcCommandOptions();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.rpc_method = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.use = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.long = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.short = reader.string();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.example = reader.string();
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.alias.push(reader.string());
                    continue;
                case 7:
                    if (tag !== 58) {
                        break;
                    }
                    message.suggest_for.push(reader.string());
                    continue;
                case 8:
                    if (tag !== 66) {
                        break;
                    }
                    message.deprecated = reader.string();
                    continue;
                case 9:
                    if (tag !== 74) {
                        break;
                    }
                    message.version = reader.string();
                    continue;
                case 10:
                    if (tag !== 82) {
                        break;
                    }
                    const entry10 = exports.RpcCommandOptions_FlagOptionsEntry.decode(reader, reader.uint32());
                    if (entry10.value !== undefined) {
                        message.flag_options[entry10.key] = entry10.value;
                    }
                    continue;
                case 11:
                    if (tag !== 90) {
                        break;
                    }
                    message.positional_args.push(exports.PositionalArgDescriptor.decode(reader, reader.uint32()));
                    continue;
                case 12:
                    if (tag !== 96) {
                        break;
                    }
                    message.skip = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            rpc_method: isSet(object.rpc_method) ? globalThis.String(object.rpc_method) : "",
            use: isSet(object.use) ? globalThis.String(object.use) : "",
            long: isSet(object.long) ? globalThis.String(object.long) : "",
            short: isSet(object.short) ? globalThis.String(object.short) : "",
            example: isSet(object.example) ? globalThis.String(object.example) : "",
            alias: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.alias) ? object.alias.map((e) => globalThis.String(e)) : [],
            suggest_for: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.suggest_for)
                ? object.suggest_for.map((e) => globalThis.String(e))
                : [],
            deprecated: isSet(object.deprecated) ? globalThis.String(object.deprecated) : "",
            version: isSet(object.version) ? globalThis.String(object.version) : "",
            flag_options: isObject(object.flag_options)
                ? Object.entries(object.flag_options).reduce((acc, [key, value]) => {
                    acc[key] = exports.FlagOptions.fromJSON(value);
                    return acc;
                }, {})
                : {},
            positional_args: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.positional_args)
                ? object.positional_args.map((e) => exports.PositionalArgDescriptor.fromJSON(e))
                : [],
            skip: isSet(object.skip) ? globalThis.Boolean(object.skip) : false,
        };
    },
    toJSON(message) {
        var _a, _b, _c;
        const obj = {};
        if (message.rpc_method !== "") {
            obj.rpc_method = message.rpc_method;
        }
        if (message.use !== "") {
            obj.use = message.use;
        }
        if (message.long !== "") {
            obj.long = message.long;
        }
        if (message.short !== "") {
            obj.short = message.short;
        }
        if (message.example !== "") {
            obj.example = message.example;
        }
        if ((_a = message.alias) === null || _a === void 0 ? void 0 : _a.length) {
            obj.alias = message.alias;
        }
        if ((_b = message.suggest_for) === null || _b === void 0 ? void 0 : _b.length) {
            obj.suggest_for = message.suggest_for;
        }
        if (message.deprecated !== "") {
            obj.deprecated = message.deprecated;
        }
        if (message.version !== "") {
            obj.version = message.version;
        }
        if (message.flag_options) {
            const entries = Object.entries(message.flag_options);
            if (entries.length > 0) {
                obj.flag_options = {};
                entries.forEach(([k, v]) => {
                    obj.flag_options[k] = exports.FlagOptions.toJSON(v);
                });
            }
        }
        if ((_c = message.positional_args) === null || _c === void 0 ? void 0 : _c.length) {
            obj.positional_args = message.positional_args.map((e) => exports.PositionalArgDescriptor.toJSON(e));
        }
        if (message.skip !== false) {
            obj.skip = message.skip;
        }
        return obj;
    },
    create(base) {
        return exports.RpcCommandOptions.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m;
        const message = createBaseRpcCommandOptions();
        message.rpc_method = (_a = object.rpc_method) !== null && _a !== void 0 ? _a : "";
        message.use = (_b = object.use) !== null && _b !== void 0 ? _b : "";
        message.long = (_c = object.long) !== null && _c !== void 0 ? _c : "";
        message.short = (_d = object.short) !== null && _d !== void 0 ? _d : "";
        message.example = (_e = object.example) !== null && _e !== void 0 ? _e : "";
        message.alias = ((_f = object.alias) === null || _f === void 0 ? void 0 : _f.map((e) => e)) || [];
        message.suggest_for = ((_g = object.suggest_for) === null || _g === void 0 ? void 0 : _g.map((e) => e)) || [];
        message.deprecated = (_h = object.deprecated) !== null && _h !== void 0 ? _h : "";
        message.version = (_j = object.version) !== null && _j !== void 0 ? _j : "";
        message.flag_options = Object.entries((_k = object.flag_options) !== null && _k !== void 0 ? _k : {}).reduce((acc, [key, value]) => {
            if (value !== undefined) {
                acc[key] = exports.FlagOptions.fromPartial(value);
            }
            return acc;
        }, {});
        message.positional_args = ((_l = object.positional_args) === null || _l === void 0 ? void 0 : _l.map((e) => exports.PositionalArgDescriptor.fromPartial(e))) || [];
        message.skip = (_m = object.skip) !== null && _m !== void 0 ? _m : false;
        return message;
    },
};
function createBaseRpcCommandOptions_FlagOptionsEntry() {
    return { key: "", value: undefined };
}
exports.RpcCommandOptions_FlagOptionsEntry = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (message.value !== undefined) {
            exports.FlagOptions.encode(message.value, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRpcCommandOptions_FlagOptionsEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.value = exports.FlagOptions.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? globalThis.String(object.key) : "",
            value: isSet(object.value) ? exports.FlagOptions.fromJSON(object.value) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key !== "") {
            obj.key = message.key;
        }
        if (message.value !== undefined) {
            obj.value = exports.FlagOptions.toJSON(message.value);
        }
        return obj;
    },
    create(base) {
        return exports.RpcCommandOptions_FlagOptionsEntry.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseRpcCommandOptions_FlagOptionsEntry();
        message.key = (_a = object.key) !== null && _a !== void 0 ? _a : "";
        message.value = (object.value !== undefined && object.value !== null)
            ? exports.FlagOptions.fromPartial(object.value)
            : undefined;
        return message;
    },
};
function createBaseFlagOptions() {
    return {
        name: "",
        shorthand: "",
        usage: "",
        default_value: "",
        deprecated: "",
        shorthand_deprecated: "",
        hidden: false,
    };
}
exports.FlagOptions = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.name !== "") {
            writer.uint32(10).string(message.name);
        }
        if (message.shorthand !== "") {
            writer.uint32(18).string(message.shorthand);
        }
        if (message.usage !== "") {
            writer.uint32(26).string(message.usage);
        }
        if (message.default_value !== "") {
            writer.uint32(34).string(message.default_value);
        }
        if (message.deprecated !== "") {
            writer.uint32(50).string(message.deprecated);
        }
        if (message.shorthand_deprecated !== "") {
            writer.uint32(58).string(message.shorthand_deprecated);
        }
        if (message.hidden !== false) {
            writer.uint32(64).bool(message.hidden);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFlagOptions();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.name = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.shorthand = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.usage = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.default_value = reader.string();
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.deprecated = reader.string();
                    continue;
                case 7:
                    if (tag !== 58) {
                        break;
                    }
                    message.shorthand_deprecated = reader.string();
                    continue;
                case 8:
                    if (tag !== 64) {
                        break;
                    }
                    message.hidden = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            name: isSet(object.name) ? globalThis.String(object.name) : "",
            shorthand: isSet(object.shorthand) ? globalThis.String(object.shorthand) : "",
            usage: isSet(object.usage) ? globalThis.String(object.usage) : "",
            default_value: isSet(object.default_value) ? globalThis.String(object.default_value) : "",
            deprecated: isSet(object.deprecated) ? globalThis.String(object.deprecated) : "",
            shorthand_deprecated: isSet(object.shorthand_deprecated) ? globalThis.String(object.shorthand_deprecated) : "",
            hidden: isSet(object.hidden) ? globalThis.Boolean(object.hidden) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.name !== "") {
            obj.name = message.name;
        }
        if (message.shorthand !== "") {
            obj.shorthand = message.shorthand;
        }
        if (message.usage !== "") {
            obj.usage = message.usage;
        }
        if (message.default_value !== "") {
            obj.default_value = message.default_value;
        }
        if (message.deprecated !== "") {
            obj.deprecated = message.deprecated;
        }
        if (message.shorthand_deprecated !== "") {
            obj.shorthand_deprecated = message.shorthand_deprecated;
        }
        if (message.hidden !== false) {
            obj.hidden = message.hidden;
        }
        return obj;
    },
    create(base) {
        return exports.FlagOptions.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e, _f, _g;
        const message = createBaseFlagOptions();
        message.name = (_a = object.name) !== null && _a !== void 0 ? _a : "";
        message.shorthand = (_b = object.shorthand) !== null && _b !== void 0 ? _b : "";
        message.usage = (_c = object.usage) !== null && _c !== void 0 ? _c : "";
        message.default_value = (_d = object.default_value) !== null && _d !== void 0 ? _d : "";
        message.deprecated = (_e = object.deprecated) !== null && _e !== void 0 ? _e : "";
        message.shorthand_deprecated = (_f = object.shorthand_deprecated) !== null && _f !== void 0 ? _f : "";
        message.hidden = (_g = object.hidden) !== null && _g !== void 0 ? _g : false;
        return message;
    },
};
function createBasePositionalArgDescriptor() {
    return { proto_field: "", varargs: false };
}
exports.PositionalArgDescriptor = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.proto_field !== "") {
            writer.uint32(10).string(message.proto_field);
        }
        if (message.varargs !== false) {
            writer.uint32(16).bool(message.varargs);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePositionalArgDescriptor();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.proto_field = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.varargs = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            proto_field: isSet(object.proto_field) ? globalThis.String(object.proto_field) : "",
            varargs: isSet(object.varargs) ? globalThis.Boolean(object.varargs) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.proto_field !== "") {
            obj.proto_field = message.proto_field;
        }
        if (message.varargs !== false) {
            obj.varargs = message.varargs;
        }
        return obj;
    },
    create(base) {
        return exports.PositionalArgDescriptor.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBasePositionalArgDescriptor();
        message.proto_field = (_a = object.proto_field) !== null && _a !== void 0 ? _a : "";
        message.varargs = (_b = object.varargs) !== null && _b !== void 0 ? _b : false;
        return message;
    },
};
function isObject(value) {
    return typeof value === "object" && value !== null;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
