// secretjs/dist/protobuf/cosmos/app/runtime/v1alpha1/module.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.178.0
//   protoc               v3.21.3
// source: cosmos/app/runtime/v1alpha1/module.proto
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoreKeyConfig = exports.Module = exports.protobufPackage = void 0;
/* eslint-disable */
const _m0 = __importStar(require("protobufjs/minimal"));
exports.protobufPackage = "cosmos.app.runtime.v1alpha1";
function createBaseModule() {
    return {
        app_name: "",
        begin_blockers: [],
        end_blockers: [],
        init_genesis: [],
        export_genesis: [],
        override_store_keys: [],
        order_migrations: [],
        precommiters: [],
        prepare_check_staters: [],
    };
}
exports.Module = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.app_name !== "") {
            writer.uint32(10).string(message.app_name);
        }
        for (const v of message.begin_blockers) {
            writer.uint32(18).string(v);
        }
        for (const v of message.end_blockers) {
            writer.uint32(26).string(v);
        }
        for (const v of message.init_genesis) {
            writer.uint32(34).string(v);
        }
        for (const v of message.export_genesis) {
            writer.uint32(42).string(v);
        }
        for (const v of message.override_store_keys) {
            exports.StoreKeyConfig.encode(v, writer.uint32(50).fork()).ldelim();
        }
        for (const v of message.order_migrations) {
            writer.uint32(58).string(v);
        }
        for (const v of message.precommiters) {
            writer.uint32(66).string(v);
        }
        for (const v of message.prepare_check_staters) {
            writer.uint32(74).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseModule();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.app_name = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.begin_blockers.push(reader.string());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.end_blockers.push(reader.string());
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.init_genesis.push(reader.string());
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.export_genesis.push(reader.string());
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.override_store_keys.push(exports.StoreKeyConfig.decode(reader, reader.uint32()));
                    continue;
                case 7:
                    if (tag !== 58) {
                        break;
                    }
                    message.order_migrations.push(reader.string());
                    continue;
                case 8:
                    if (tag !== 66) {
                        break;
                    }
                    message.precommiters.push(reader.string());
                    continue;
                case 9:
                    if (tag !== 74) {
                        break;
                    }
                    message.prepare_check_staters.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            app_name: isSet(object.app_name) ? globalThis.String(object.app_name) : "",
            begin_blockers: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.begin_blockers)
                ? object.begin_blockers.map((e) => globalThis.String(e))
                : [],
            end_blockers: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.end_blockers)
                ? object.end_blockers.map((e) => globalThis.String(e))
                : [],
            init_genesis: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.init_genesis)
                ? object.init_genesis.map((e) => globalThis.String(e))
                : [],
            export_genesis: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.export_genesis)
                ? object.export_genesis.map((e) => globalThis.String(e))
                : [],
            override_store_keys: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.override_store_keys)
                ? object.override_store_keys.map((e) => exports.StoreKeyConfig.fromJSON(e))
                : [],
            order_migrations: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.order_migrations)
                ? object.order_migrations.map((e) => globalThis.String(e))
                : [],
            precommiters: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.precommiters)
                ? object.precommiters.map((e) => globalThis.String(e))
                : [],
            prepare_check_staters: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.prepare_check_staters)
                ? object.prepare_check_staters.map((e) => globalThis.String(e))
                : [],
        };
    },
    toJSON(message) {
        var _a, _b, _c, _d, _e, _f, _g, _h;
        const obj = {};
        if (message.app_name !== "") {
            obj.app_name = message.app_name;
        }
        if ((_a = message.begin_blockers) === null || _a === void 0 ? void 0 : _a.length) {
            obj.begin_blockers = message.begin_blockers;
        }
        if ((_b = message.end_blockers) === null || _b === void 0 ? void 0 : _b.length) {
            obj.end_blockers = message.end_blockers;
        }
        if ((_c = message.init_genesis) === null || _c === void 0 ? void 0 : _c.length) {
            obj.init_genesis = message.init_genesis;
        }
        if ((_d = message.export_genesis) === null || _d === void 0 ? void 0 : _d.length) {
            obj.export_genesis = message.export_genesis;
        }
        if ((_e = message.override_store_keys) === null || _e === void 0 ? void 0 : _e.length) {
            obj.override_store_keys = message.override_store_keys.map((e) => exports.StoreKeyConfig.toJSON(e));
        }
        if ((_f = message.order_migrations) === null || _f === void 0 ? void 0 : _f.length) {
            obj.order_migrations = message.order_migrations;
        }
        if ((_g = message.precommiters) === null || _g === void 0 ? void 0 : _g.length) {
            obj.precommiters = message.precommiters;
        }
        if ((_h = message.prepare_check_staters) === null || _h === void 0 ? void 0 : _h.length) {
            obj.prepare_check_staters = message.prepare_check_staters;
        }
        return obj;
    },
    create(base) {
        return exports.Module.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j;
        const message = createBaseModule();
        message.app_name = (_a = object.app_name) !== null && _a !== void 0 ? _a : "";
        message.begin_blockers = ((_b = object.begin_blockers) === null || _b === void 0 ? void 0 : _b.map((e) => e)) || [];
        message.end_blockers = ((_c = object.end_blockers) === null || _c === void 0 ? void 0 : _c.map((e) => e)) || [];
        message.init_genesis = ((_d = object.init_genesis) === null || _d === void 0 ? void 0 : _d.map((e) => e)) || [];
        message.export_genesis = ((_e = object.export_genesis) === null || _e === void 0 ? void 0 : _e.map((e) => e)) || [];
        message.override_store_keys = ((_f = object.override_store_keys) === null || _f === void 0 ? void 0 : _f.map((e) => exports.StoreKeyConfig.fromPartial(e))) || [];
        message.order_migrations = ((_g = object.order_migrations) === null || _g === void 0 ? void 0 : _g.map((e) => e)) || [];
        message.precommiters = ((_h = object.precommiters) === null || _h === void 0 ? void 0 : _h.map((e) => e)) || [];
        message.prepare_check_staters = ((_j = object.prepare_check_staters) === null || _j === void 0 ? void 0 : _j.map((e) => e)) || [];
        return message;
    },
};
function createBaseStoreKeyConfig() {
    return { module_name: "", kv_store_key: "" };
}
exports.StoreKeyConfig = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.module_name !== "") {
            writer.uint32(10).string(message.module_name);
        }
        if (message.kv_store_key !== "") {
            writer.uint32(18).string(message.kv_store_key);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseStoreKeyConfig();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.module_name = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.kv_store_key = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            module_name: isSet(object.module_name) ? globalThis.String(object.module_name) : "",
            kv_store_key: isSet(object.kv_store_key) ? globalThis.String(object.kv_store_key) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.module_name !== "") {
            obj.module_name = message.module_name;
        }
        if (message.kv_store_key !== "") {
            obj.kv_store_key = message.kv_store_key;
        }
        return obj;
    },
    create(base) {
        return exports.StoreKeyConfig.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseStoreKeyConfig();
        message.module_name = (_a = object.module_name) !== null && _a !== void 0 ? _a : "";
        message.kv_store_key = (_b = object.kv_store_key) !== null && _b !== void 0 ? _b : "";
        return message;
    },
};
function isSet(value) {
    return value !== null && value !== undefined;
}
