// secretjs/dist/protobuf/cosmos/slashing/v1beta1/slashing.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.178.0
//   protoc               v3.21.3
// source: cosmos/slashing/v1beta1/slashing.proto
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Params = exports.ValidatorSigningInfo = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const _m0 = __importStar(require("protobufjs/minimal"));
const duration_1 = require("../../../google/protobuf/duration");
const timestamp_1 = require("../../../google/protobuf/timestamp");
exports.protobufPackage = "cosmos.slashing.v1beta1";
function createBaseValidatorSigningInfo() {
    return {
        address: "",
        start_height: "0",
        index_offset: "0",
        jailed_until: undefined,
        tombstoned: false,
        missed_blocks_counter: "0",
    };
}
exports.ValidatorSigningInfo = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.start_height !== "0") {
            writer.uint32(16).int64(message.start_height);
        }
        if (message.index_offset !== "0") {
            writer.uint32(24).int64(message.index_offset);
        }
        if (message.jailed_until !== undefined) {
            timestamp_1.Timestamp.encode(message.jailed_until, writer.uint32(34).fork()).ldelim();
        }
        if (message.tombstoned !== false) {
            writer.uint32(40).bool(message.tombstoned);
        }
        if (message.missed_blocks_counter !== "0") {
            writer.uint32(48).int64(message.missed_blocks_counter);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidatorSigningInfo();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.start_height = longToString(reader.int64());
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.index_offset = longToString(reader.int64());
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.jailed_until = timestamp_1.Timestamp.decode(reader, reader.uint32());
                    continue;
                case 5:
                    if (tag !== 40) {
                        break;
                    }
                    message.tombstoned = reader.bool();
                    continue;
                case 6:
                    if (tag !== 48) {
                        break;
                    }
                    message.missed_blocks_counter = longToString(reader.int64());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            address: isSet(object.address) ? globalThis.String(object.address) : "",
            start_height: isSet(object.start_height) ? globalThis.String(object.start_height) : "0",
            index_offset: isSet(object.index_offset) ? globalThis.String(object.index_offset) : "0",
            jailed_until: isSet(object.jailed_until) ? fromJsonTimestamp(object.jailed_until) : undefined,
            tombstoned: isSet(object.tombstoned) ? globalThis.Boolean(object.tombstoned) : false,
            missed_blocks_counter: isSet(object.missed_blocks_counter)
                ? globalThis.String(object.missed_blocks_counter)
                : "0",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.start_height !== "0") {
            obj.start_height = message.start_height;
        }
        if (message.index_offset !== "0") {
            obj.index_offset = message.index_offset;
        }
        if (message.jailed_until !== undefined) {
            obj.jailed_until = fromTimestamp(message.jailed_until).toISOString();
        }
        if (message.tombstoned !== false) {
            obj.tombstoned = message.tombstoned;
        }
        if (message.missed_blocks_counter !== "0") {
            obj.missed_blocks_counter = message.missed_blocks_counter;
        }
        return obj;
    },
    create(base) {
        return exports.ValidatorSigningInfo.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e;
        const message = createBaseValidatorSigningInfo();
        message.address = (_a = object.address) !== null && _a !== void 0 ? _a : "";
        message.start_height = (_b = object.start_height) !== null && _b !== void 0 ? _b : "0";
        message.index_offset = (_c = object.index_offset) !== null && _c !== void 0 ? _c : "0";
        message.jailed_until = (object.jailed_until !== undefined && object.jailed_until !== null)
            ? timestamp_1.Timestamp.fromPartial(object.jailed_until)
            : undefined;
        message.tombstoned = (_d = object.tombstoned) !== null && _d !== void 0 ? _d : false;
        message.missed_blocks_counter = (_e = object.missed_blocks_counter) !== null && _e !== void 0 ? _e : "0";
        return message;
    },
};
function createBaseParams() {
    return {
        signed_blocks_window: "0",
        min_signed_per_window: new Uint8Array(0),
        downtime_jail_duration: undefined,
        slash_fraction_double_sign: new Uint8Array(0),
        slash_fraction_downtime: new Uint8Array(0),
    };
}
exports.Params = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.signed_blocks_window !== "0") {
            writer.uint32(8).int64(message.signed_blocks_window);
        }
        if (message.min_signed_per_window.length !== 0) {
            writer.uint32(18).bytes(message.min_signed_per_window);
        }
        if (message.downtime_jail_duration !== undefined) {
            duration_1.Duration.encode(message.downtime_jail_duration, writer.uint32(26).fork()).ldelim();
        }
        if (message.slash_fraction_double_sign.length !== 0) {
            writer.uint32(34).bytes(message.slash_fraction_double_sign);
        }
        if (message.slash_fraction_downtime.length !== 0) {
            writer.uint32(42).bytes(message.slash_fraction_downtime);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.signed_blocks_window = longToString(reader.int64());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.min_signed_per_window = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.downtime_jail_duration = duration_1.Duration.decode(reader, reader.uint32());
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.slash_fraction_double_sign = reader.bytes();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.slash_fraction_downtime = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            signed_blocks_window: isSet(object.signed_blocks_window) ? globalThis.String(object.signed_blocks_window) : "0",
            min_signed_per_window: isSet(object.min_signed_per_window)
                ? bytesFromBase64(object.min_signed_per_window)
                : new Uint8Array(0),
            downtime_jail_duration: isSet(object.downtime_jail_duration)
                ? duration_1.Duration.fromJSON(object.downtime_jail_duration)
                : undefined,
            slash_fraction_double_sign: isSet(object.slash_fraction_double_sign)
                ? bytesFromBase64(object.slash_fraction_double_sign)
                : new Uint8Array(0),
            slash_fraction_downtime: isSet(object.slash_fraction_downtime)
                ? bytesFromBase64(object.slash_fraction_downtime)
                : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.signed_blocks_window !== "0") {
            obj.signed_blocks_window = message.signed_blocks_window;
        }
        if (message.min_signed_per_window.length !== 0) {
            obj.min_signed_per_window = base64FromBytes(message.min_signed_per_window);
        }
        if (message.downtime_jail_duration !== undefined) {
            obj.downtime_jail_duration = duration_1.Duration.toJSON(message.downtime_jail_duration);
        }
        if (message.slash_fraction_double_sign.length !== 0) {
            obj.slash_fraction_double_sign = base64FromBytes(message.slash_fraction_double_sign);
        }
        if (message.slash_fraction_downtime.length !== 0) {
            obj.slash_fraction_downtime = base64FromBytes(message.slash_fraction_downtime);
        }
        return obj;
    },
    create(base) {
        return exports.Params.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d;
        const message = createBaseParams();
        message.signed_blocks_window = (_a = object.signed_blocks_window) !== null && _a !== void 0 ? _a : "0";
        message.min_signed_per_window = (_b = object.min_signed_per_window) !== null && _b !== void 0 ? _b : new Uint8Array(0);
        message.downtime_jail_duration =
            (object.downtime_jail_duration !== undefined && object.downtime_jail_duration !== null)
                ? duration_1.Duration.fromPartial(object.downtime_jail_duration)
                : undefined;
        message.slash_fraction_double_sign = (_c = object.slash_fraction_double_sign) !== null && _c !== void 0 ? _c : new Uint8Array(0);
        message.slash_fraction_downtime = (_d = object.slash_fraction_downtime) !== null && _d !== void 0 ? _d : new Uint8Array(0);
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
function toTimestamp(date) {
    const seconds = Math.trunc(date.getTime() / 1000).toString();
    const nanos = (date.getTime() % 1000) * 1000000;
    return { seconds, nanos };
}
function fromTimestamp(t) {
    let millis = (globalThis.Number(t.seconds) || 0) * 1000;
    millis += (t.nanos || 0) / 1000000;
    return new globalThis.Date(millis);
}
function fromJsonTimestamp(o) {
    if (o instanceof globalThis.Date) {
        return toTimestamp(o);
    }
    else if (typeof o === "string") {
        return toTimestamp(new globalThis.Date(o));
    }
    else {
        return timestamp_1.Timestamp.fromJSON(o);
    }
}
function longToString(long) {
    return long.toString();
}
if (_m0.util.Long !== long_1.default) {
    _m0.util.Long = long_1.default;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
