// secretjs/dist/tx/bank.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgSetSendEnabled = exports.MsgMultiSend = exports.MsgSend = exports.MsgSetSendEnabledParams = exports.MsgMultiSendParams = exports.MsgSendParams = void 0;
const tx_1 = require("../protobuf/cosmos/bank/v1beta1/tx");
var tx_2 = require("../protobuf/cosmos/bank/v1beta1/tx");
Object.defineProperty(exports, "MsgSendParams", { enumerable: true, get: function () { return tx_2.MsgSend; } });
Object.defineProperty(exports, "MsgMultiSendParams", { enumerable: true, get: function () { return tx_2.MsgMultiSend; } });
Object.defineProperty(exports, "MsgSetSendEnabledParams", { enumerable: true, get: function () { return tx_2.MsgSetSendEnabled; } });
/** MsgSend represents a message to send coins from one account to another. */
class MsgSend {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/cosmos.bank.v1beta1.MsgSend",
                value: this.params,
                encode: () => tx_1.MsgSend.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgSend",
                value: this.params,
            };
        });
    }
}
exports.MsgSend = MsgSend;
/** MsgMultiSend represents an arbitrary multi-in, multi-out send message. */
class MsgMultiSend {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/cosmos.bank.v1beta1.MsgMultiSend",
                value: this.params,
                encode: () => tx_1.MsgMultiSend.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgMultiSend",
                value: this.params,
            };
        });
    }
}
exports.MsgMultiSend = MsgMultiSend;
class MsgSetSendEnabled {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/cosmos.bank.v1beta1.MsgSetSendEnabled",
                value: this.params,
                encode: () => tx_1.MsgSetSendEnabled.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgSetSendEnabled",
                value: this.params,
            };
        });
    }
}
exports.MsgSetSendEnabled = MsgSetSendEnabled;
