// secretjs/dist/tx/crisis.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgVerifyInvariant = exports.MsgVerifyInvariantParams = void 0;
const tx_1 = require("../protobuf/cosmos/crisis/v1beta1/tx");
var tx_2 = require("../protobuf/cosmos/crisis/v1beta1/tx");
Object.defineProperty(exports, "MsgVerifyInvariantParams", { enumerable: true, get: function () { return tx_2.MsgVerifyInvariant; } });
/** MsgVerifyInvariant represents a message to verify a particular invariance. */
class MsgVerifyInvariant {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/cosmos.crisis.v1beta1.MsgVerifyInvariant",
                value: this.params,
                encode: () => tx_1.MsgVerifyInvariant.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgVerifyInvariant",
                value: this.params,
            };
        });
    }
}
exports.MsgVerifyInvariant = MsgVerifyInvariant;
