// secretjs/dist/tx/distribution.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgDepositValidatorRewardsPool = exports.MsgCommunityPoolSpend = exports.MsgSetAutoRestake = exports.MsgFundCommunityPool = exports.MsgWithdrawValidatorCommission = exports.MsgWithdrawDelegatorReward = exports.MsgSetWithdrawAddress = exports.MsgDepositValidatorRewardsPoolParams = exports.MsgCommunityPoolSpendParams = exports.MsgSetAutoRestakeParams = exports.MsgFundCommunityPoolParams = exports.MsgWithdrawValidatorCommissionParams = exports.MsgWithdrawDelegatorRewardParams = exports.MsgSetWithdrawAddressParams = void 0;
const encoding_1 = require("@cosmjs/encoding");
const tx_1 = require("../protobuf/cosmos/distribution/v1beta1/tx");
var tx_2 = require("../protobuf/cosmos/distribution/v1beta1/tx");
Object.defineProperty(exports, "MsgSetWithdrawAddressParams", { enumerable: true, get: function () { return tx_2.MsgSetWithdrawAddress; } });
Object.defineProperty(exports, "MsgWithdrawDelegatorRewardParams", { enumerable: true, get: function () { return tx_2.MsgWithdrawDelegatorReward; } });
Object.defineProperty(exports, "MsgWithdrawValidatorCommissionParams", { enumerable: true, get: function () { return tx_2.MsgWithdrawValidatorCommission; } });
Object.defineProperty(exports, "MsgFundCommunityPoolParams", { enumerable: true, get: function () { return tx_2.MsgFundCommunityPool; } });
Object.defineProperty(exports, "MsgSetAutoRestakeParams", { enumerable: true, get: function () { return tx_2.MsgSetAutoRestake; } });
Object.defineProperty(exports, "MsgCommunityPoolSpendParams", { enumerable: true, get: function () { return tx_2.MsgCommunityPoolSpend; } });
Object.defineProperty(exports, "MsgDepositValidatorRewardsPoolParams", { enumerable: true, get: function () { return tx_2.MsgDepositValidatorRewardsPool; } });
/**
 * MsgSetWithdrawAddress sets the withdraw address for
 * a delegator (or validator self-delegation).
 */
class MsgSetWithdrawAddress {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/cosmos.distribution.v1beta1.MsgSetWithdrawAddress",
                value: this.params,
                encode: () => tx_1.MsgSetWithdrawAddress.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgModifyWithdrawAddress",
                value: this.params,
            };
        });
    }
}
exports.MsgSetWithdrawAddress = MsgSetWithdrawAddress;
/**
 * MsgWithdrawDelegatorReward represents delegation withdrawal to a delegator
 * from a single validator.
 */
class MsgWithdrawDelegatorReward {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward",
                value: this.params,
                encode: () => tx_1.MsgWithdrawDelegatorReward.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgWithdrawDelegationReward",
                value: this.params,
            };
        });
    }
}
exports.MsgWithdrawDelegatorReward = MsgWithdrawDelegatorReward;
/**
 * MsgWithdrawValidatorCommission withdraws the full commission to the validator
 * address.
 */
class MsgWithdrawValidatorCommission {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/cosmos.distribution.v1beta1.MsgWithdrawValidatorCommission",
                value: this.params,
                encode: () => tx_1.MsgWithdrawValidatorCommission.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgWithdrawValCommission",
                value: this.params,
            };
        });
    }
}
exports.MsgWithdrawValidatorCommission = MsgWithdrawValidatorCommission;
/**
 * MsgFundCommunityPool allows an account to directly
 * fund the community pool.
 */
class MsgFundCommunityPool {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/cosmos.distribution.v1beta1.MsgFundCommunityPool",
                value: this.params,
                encode: () => tx_1.MsgFundCommunityPool.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgFundCommunityPool",
                value: this.params,
            };
        });
    }
}
exports.MsgFundCommunityPool = MsgFundCommunityPool;
/**
 * MsgSetAutoRestake enables or disables auto-restaking for
 * a delegator-validator pair.
 */
class MsgSetAutoRestake {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/cosmos.distribution.v1beta1.MsgSetAutoRestake",
                value: this.params,
                encode: () => tx_1.MsgSetAutoRestake.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgSetAutoRestake",
                value: Object.assign({}, this.params, {
                    enabled: this.params.enabled ? true : undefined,
                }),
            };
        });
    }
}
exports.MsgSetAutoRestake = MsgSetAutoRestake;
class MsgCommunityPoolSpend {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/cosmos.distribution.v1beta1.MsgCommunityPoolSpend",
                value: this.params,
                encode: () => tx_1.MsgCommunityPoolSpend.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgCommunityPoolSpend",
                value: this.params,
            };
        });
    }
}
exports.MsgCommunityPoolSpend = MsgCommunityPoolSpend;
class MsgDepositValidatorRewardsPool {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            const msgContent = Object.assign(Object.assign({}, this.params), { validator_address: (0, encoding_1.toBech32)("secretvaloper", (0, encoding_1.fromBech32)(this.params.validator_address).data) });
            return {
                type_url: "/cosmos.distribution.v1beta1.MsgDepositValidatorRewardsPool",
                value: msgContent,
                encode: () => tx_1.MsgDepositValidatorRewardsPool.encode(msgContent).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgDepositValidatorRewardsPool",
                value: this.params,
            };
        });
    }
}
exports.MsgDepositValidatorRewardsPool = MsgDepositValidatorRewardsPool;
