// secretjs/dist/tx/emergency_button.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgUpdateParams = exports.MsgToggleIbcSwitch = exports.EmergencyButtonParams = exports.MsgUpdateParamsParams = exports.MsgToggleIbcSwitchParams = void 0;
const tx_1 = require("../protobuf/secret/emergencybutton/v1beta1/tx");
var tx_2 = require("../protobuf/secret/emergencybutton/v1beta1/tx");
Object.defineProperty(exports, "MsgToggleIbcSwitchParams", { enumerable: true, get: function () { return tx_2.MsgToggleIbcSwitch; } });
Object.defineProperty(exports, "MsgUpdateParamsParams", { enumerable: true, get: function () { return tx_2.MsgUpdateParams; } });
var params_1 = require("../protobuf/secret/emergencybutton/v1beta1/params");
Object.defineProperty(exports, "EmergencyButtonParams", { enumerable: true, get: function () { return params_1.Params; } });
class MsgToggleIbcSwitch {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/secret.emergencybutton.v1beta1.MsgToggleIbcSwitch",
                value: this.params,
                encode: () => tx_1.MsgToggleIbcSwitch.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "emergencybutton/MsgToggleIbcSwitch",
                value: this.params,
            };
        });
    }
}
exports.MsgToggleIbcSwitch = MsgToggleIbcSwitch;
class MsgUpdateParams {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/secret.emergencybutton.v1beta1.MsgUpdateParams",
                value: this.params,
                encode: () => tx_1.MsgUpdateParams.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "emergencybutton/MsgUpdateParams",
                value: this.params,
            };
        });
    }
}
exports.MsgUpdateParams = MsgUpdateParams;
