// secretjs/dist/tx/ibc_transfer.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgTransfer = void 0;
const tx_1 = require("../protobuf/ibc/applications/transfer/v1/tx");
/**
 * MsgTransfer defines a msg to transfer fungible tokens (i.e Coins) between
 * ICS20 enabled chains. See ICS Spec here:
 * https://github.com/cosmos/ics/tree/master/spec/ics-020-fungible-token-transfer#data-structures
 */
class MsgTransfer {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const msgContent = {
                source_port: this.params.source_port,
                source_channel: this.params.source_channel,
                token: this.params.token,
                sender: this.params.sender,
                receiver: this.params.receiver,
                timeout_height: this.params.timeout_height,
                timeout_timestamp: this.params.timeout_timestamp
                    ? `${this.params.timeout_timestamp}000000000` // sec -> ns
                    : "0",
                memo: (_a = this.params.memo) !== null && _a !== void 0 ? _a : "",
            };
            return {
                type_url: "/ibc.applications.transfer.v1.MsgTransfer",
                value: msgContent,
                encode: () => tx_1.MsgTransfer.encode(msgContent).finish(),
            };
        });
    }
    toAmino() {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgTransfer",
                value: {
                    source_port: this.params.source_port,
                    source_channel: this.params.source_channel,
                    token: this.params.token,
                    sender: this.params.sender,
                    receiver: this.params.receiver,
                    timeout_height: this.params.timeout_height
                        ? {
                            revision_number: this.params.timeout_height.revision_number,
                            revision_height: this.params.timeout_height.revision_height,
                        }
                        : {},
                    timeout_timestamp: this.params.timeout_timestamp
                        ? `${this.params.timeout_timestamp}000000000` // sec -> ns
                        : "0",
                    memo: (_a = this.params.memo) !== null && _a !== void 0 ? _a : "",
                },
            };
        });
    }
}
exports.MsgTransfer = MsgTransfer;
