// secretjs/dist/tx/ibc_client.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgIBCSoftwareUpgrade = exports.MsgRecoverClient = exports.MsgCreateClient = exports.MsgSubmitMisbehaviour = exports.MsgUpgradeClient = exports.MsgUpdateClient = exports.MsgIBCSoftwareUpgradeParams = exports.MsgRecoverClientParams = exports.MsgSubmitMisbehaviourParams = exports.MsgUpgradeClientParams = exports.MsgUpdateClientParams = exports.MsgCreateClientParams = void 0;
const tx_1 = require("../protobuf/ibc/core/client/v1/tx");
var tx_2 = require("../protobuf/ibc/core/client/v1/tx");
Object.defineProperty(exports, "MsgCreateClientParams", { enumerable: true, get: function () { return tx_2.MsgCreateClient; } });
Object.defineProperty(exports, "MsgUpdateClientParams", { enumerable: true, get: function () { return tx_2.MsgUpdateClient; } });
Object.defineProperty(exports, "MsgUpgradeClientParams", { enumerable: true, get: function () { return tx_2.MsgUpgradeClient; } });
Object.defineProperty(exports, "MsgSubmitMisbehaviourParams", { enumerable: true, get: function () { return tx_2.MsgSubmitMisbehaviour; } });
Object.defineProperty(exports, "MsgRecoverClientParams", { enumerable: true, get: function () { return tx_2.MsgRecoverClient; } });
Object.defineProperty(exports, "MsgIBCSoftwareUpgradeParams", { enumerable: true, get: function () { return tx_2.MsgIBCSoftwareUpgrade; } });
/** MsgUpdateClient defines an sdk.Msg to update a IBC client state using the given header. */
class MsgUpdateClient {
    constructor(msg) {
        this.msg = msg;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/ibc.core.client.v1.MsgUpdateClient",
                value: this.msg,
                encode: () => tx_1.MsgUpdateClient.encode(this.msg).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            throw new Error("MsgUpdateClient doesn't support Amino encoding.");
        });
    }
}
exports.MsgUpdateClient = MsgUpdateClient;
/** MsgUpdateClient defines an sdk.Msg to update a IBC client state using the given header. */
class MsgUpgradeClient {
    constructor(msg) {
        this.msg = msg;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/ibc.core.client.v1.MsgUpgradeClient",
                value: this.msg,
                encode: () => tx_1.MsgUpgradeClient.encode(this.msg).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            throw new Error("MsgUpgradeClient doesn't support Amino encoding.");
        });
    }
}
exports.MsgUpgradeClient = MsgUpgradeClient;
/** MsgSubmitMisbehaviour defines an sdk.Msg type that submits Evidence for light client misbehaviour. */
class MsgSubmitMisbehaviour {
    constructor(msg) {
        this.msg = msg;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/ibc.core.client.v1.MsgSubmitMisbehaviour",
                value: this.msg,
                encode: () => tx_1.MsgSubmitMisbehaviour.encode(this.msg).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            throw new Error("MsgSubmitMisbehaviour doesn't support Amino encoding.");
        });
    }
}
exports.MsgSubmitMisbehaviour = MsgSubmitMisbehaviour;
/** MsgCreateClient defines a message to create an IBC client */
class MsgCreateClient {
    constructor(msg) {
        this.msg = msg;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/ibc.core.client.v1.MsgCreateClient",
                value: this.msg,
                encode: () => tx_1.MsgCreateClient.encode(this.msg).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            throw new Error("MsgCreateClient doesn't support Amino encoding.");
        });
    }
}
exports.MsgCreateClient = MsgCreateClient;
class MsgRecoverClient {
    constructor(msg) {
        this.msg = msg;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/ibc.core.client.v1.MsgRecoverClient",
                value: this.msg,
                encode: () => tx_1.MsgRecoverClient.encode(this.msg).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            throw new Error("MsgRecoverClient doesn't support Amino encoding.");
        });
    }
}
exports.MsgRecoverClient = MsgRecoverClient;
class MsgIBCSoftwareUpgrade {
    constructor(msg) {
        this.msg = msg;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/ibc.core.client.v1.MsgIBCSoftwareUpgrade",
                value: this.msg,
                encode: () => tx_1.MsgIBCSoftwareUpgrade.encode(this.msg).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            throw new Error("MsgIBCSoftwareUpgrade doesn't support Amino encoding.");
        });
    }
}
exports.MsgIBCSoftwareUpgrade = MsgIBCSoftwareUpgrade;
