// secretjs/dist/tx/ibc_interchain_accounts.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgSendTx = exports.MsgRegisterInterchainAccount = exports.MsgSendTxParams = exports.MsgRegisterInterchainAccountParams = void 0;
const tx_1 = require("../protobuf/ibc/applications/interchain_accounts/controller/v1/tx");
var tx_2 = require("../protobuf/ibc/applications/interchain_accounts/controller/v1/tx");
Object.defineProperty(exports, "MsgRegisterInterchainAccountParams", { enumerable: true, get: function () { return tx_2.MsgRegisterInterchainAccount; } });
Object.defineProperty(exports, "MsgSendTxParams", { enumerable: true, get: function () { return tx_2.MsgSendTx; } });
class MsgRegisterInterchainAccount {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: `/ibc.applications.interchain_accounts.controller.v1.MsgRegisterInterchainAccount`,
                value: this.params,
                encode: () => tx_1.MsgRegisterInterchainAccount.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            throw new Error("Not implemented. Please use WalletProto");
        });
    }
}
exports.MsgRegisterInterchainAccount = MsgRegisterInterchainAccount;
class MsgSendTx {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: `/ibc.applications.interchain_accounts.controller.v1.MsgSendTx`,
                value: this.params,
                encode: () => tx_1.MsgSendTx.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            throw new Error("Not implemented. Please use WalletProto");
        });
    }
}
exports.MsgSendTx = MsgSendTx;
