// secretjs/dist/tx/staking.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgCancelUnbondingDelegation = exports.MsgUndelegate = exports.MsgBeginRedelegate = exports.MsgDelegate = exports.MsgEditValidator = exports.MsgCreateValidator = exports.MsgCancelUnbondingDelegationParams = exports.MsgUndelegateParams = exports.MsgBeginRedelegateParams = exports.MsgDelegateParams = exports.ValidatorDescription = void 0;
const encoding_1 = require("@cosmjs/encoding");
const bignumber_js_1 = __importDefault(require("bignumber.js"));
const keys_1 = require("../protobuf/cosmos/crypto/ed25519/keys");
const any_1 = require("../protobuf/google/protobuf/any");
const staking_1 = require("../protobuf/cosmos/staking/v1beta1/staking");
const tx_1 = require("../protobuf/cosmos/staking/v1beta1/tx");
var staking_2 = require("../protobuf/cosmos/staking/v1beta1/staking");
Object.defineProperty(exports, "ValidatorDescription", { enumerable: true, get: function () { return staking_2.Description; } });
var tx_2 = require("../protobuf/cosmos/staking/v1beta1/tx");
Object.defineProperty(exports, "MsgDelegateParams", { enumerable: true, get: function () { return tx_2.MsgDelegate; } });
Object.defineProperty(exports, "MsgBeginRedelegateParams", { enumerable: true, get: function () { return tx_2.MsgBeginRedelegate; } });
Object.defineProperty(exports, "MsgUndelegateParams", { enumerable: true, get: function () { return tx_2.MsgUndelegate; } });
Object.defineProperty(exports, "MsgCancelUnbondingDelegationParams", { enumerable: true, get: function () { return tx_2.MsgCancelUnbondingDelegation; } });
/** MsgCreateValidator defines an SDK message for creating a new validator. */
class MsgCreateValidator {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            const msgContent = {
                description: this.params.description,
                commission: {
                    rate: new bignumber_js_1.default(this.params.commission.rate)
                        .toFixed(18)
                        .replace(/0\.0*/, ""),
                    max_rate: new bignumber_js_1.default(this.params.commission.max_rate)
                        .toFixed(18)
                        .replace(/0\.0*/, ""),
                    max_change_rate: new bignumber_js_1.default(this.params.commission.max_change_rate)
                        .toFixed(18)
                        .replace(/0\.0*/, ""),
                },
                min_self_delegation: this.params.min_self_delegation,
                delegator_address: this.params.delegator_address,
                validator_address: (0, encoding_1.toBech32)("secretvaloper", (0, encoding_1.fromBech32)(this.params.delegator_address).data),
                pubkey: any_1.Any.fromPartial({
                    type_url: "/cosmos.crypto.ed25519.PubKey",
                    value: keys_1.PubKey.encode(keys_1.PubKey.fromPartial({
                        key: (0, encoding_1.fromBase64)(this.params.pubkey),
                    })).finish(),
                }),
                value: this.params.initial_delegation,
            };
            return {
                type_url: `/cosmos.staking.v1beta1.MsgCreateValidator`,
                value: msgContent,
                encode: () => tx_1.MsgCreateValidator.encode(msgContent).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgCreateValidator",
                value: {
                    description: {
                        moniker: this.params.description.moniker,
                        identity: this.params.description.identity,
                        website: this.params.description.website,
                        security_contact: this.params.description.security_contact,
                        details: this.params.description.details,
                    },
                    commission: {
                        rate: new bignumber_js_1.default(this.params.commission.rate)
                            .toFixed(18)
                            .replace(/0\.0*/, ""),
                        max_rate: new bignumber_js_1.default(this.params.commission.max_rate)
                            .toFixed(18)
                            .replace(/0\.0*/, ""),
                        max_change_rate: new bignumber_js_1.default(this.params.commission.max_change_rate)
                            .toFixed(18)
                            .replace(/0\.0*/, ""),
                    },
                    min_self_delegation: this.params.min_self_delegation,
                    delegator_address: this.params.delegator_address,
                    validator_address: (0, encoding_1.toBech32)("secretvaloper", (0, encoding_1.fromBech32)(this.params.delegator_address).data),
                    pubkey: {
                        type: "tendermint/PubKeyEd25519",
                        value: this.params.pubkey,
                    },
                    value: this.params.initial_delegation,
                },
            };
        });
    }
}
exports.MsgCreateValidator = MsgCreateValidator;
/** MsgEditValidator defines an SDK message for editing an existing validator. */
class MsgEditValidator {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            const msgContent = {
                validator_address: this.params.validator_address,
                description: staking_1.Description.fromPartial(this.params.description || {}),
                commission_rate: this.params.commission_rate
                    ? new bignumber_js_1.default(this.params.commission_rate)
                        .toFixed(18)
                        .replace(/0\.0*/, "")
                    : "",
                min_self_delegation: this.params.min_self_delegation || "",
            };
            return {
                type_url: `/cosmos.staking.v1beta1.MsgEditValidator`,
                value: msgContent,
                encode: () => tx_1.MsgEditValidator.encode(msgContent).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            let description;
            if (this.params.description) {
                description = {
                    moniker: this.params.description.moniker,
                    identity: this.params.description.identity,
                    website: this.params.description.website,
                    security_contact: this.params.description.security_contact,
                    details: this.params.description.details,
                };
            }
            let commission_rate;
            if (this.params.commission_rate) {
                commission_rate = new bignumber_js_1.default(this.params.commission_rate).toFixed(18);
            }
            return {
                type: "cosmos-sdk/MsgEditValidator",
                value: {
                    validator_address: this.params.validator_address,
                    description,
                    commission_rate,
                    min_self_delegation: this.params.min_self_delegation,
                },
            };
        });
    }
}
exports.MsgEditValidator = MsgEditValidator;
/** MsgDelegate defines an SDK message for performing a delegation of coins from a delegator to a validator. */
class MsgDelegate {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: `/cosmos.staking.v1beta1.MsgDelegate`,
                value: this.params,
                encode: () => tx_1.MsgDelegate.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgDelegate",
                value: this.params,
            };
        });
    }
}
exports.MsgDelegate = MsgDelegate;
/** MsgBeginRedelegate defines an SDK message for performing a redelegation of coins from a delegator and source validator to a destination validator. */
class MsgBeginRedelegate {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: `/cosmos.staking.v1beta1.MsgBeginRedelegate`,
                value: this.params,
                encode: () => tx_1.MsgBeginRedelegate.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgBeginRedelegate",
                value: this.params,
            };
        });
    }
}
exports.MsgBeginRedelegate = MsgBeginRedelegate;
/** MsgUndelegate defines an SDK message for performing an undelegation from a delegate and a validator */
class MsgUndelegate {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: `/cosmos.staking.v1beta1.MsgUndelegate`,
                value: this.params,
                encode: () => tx_1.MsgUndelegate.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgUndelegate",
                value: this.params,
            };
        });
    }
}
exports.MsgUndelegate = MsgUndelegate;
class MsgCancelUnbondingDelegation {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: `/cosmos.staking.v1beta1.MsgCancelUnbondingDelegation`,
                value: this.params,
                encode: () => tx_1.MsgCancelUnbondingDelegation.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type: "cosmos-sdk/MsgCancelUnbondingDelegation",
                value: this.params,
            };
        });
    }
}
exports.MsgCancelUnbondingDelegation = MsgCancelUnbondingDelegation;
