// secretjs/dist/tx/vesting.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgCreatePeriodicVestingAccount = exports.MsgCreatePermanentLockedAccount = exports.MsgCreateVestingAccount = exports.MsgCreatePeriodicVestingAccountParams = exports.MsgCreatePermanentLockedAccountParams = exports.MsgCreateVestingAccountParams = void 0;
const tx_1 = require("../protobuf/cosmos/vesting/v1beta1/tx");
var tx_2 = require("../protobuf/cosmos/vesting/v1beta1/tx");
Object.defineProperty(exports, "MsgCreateVestingAccountParams", { enumerable: true, get: function () { return tx_2.MsgCreateVestingAccount; } });
Object.defineProperty(exports, "MsgCreatePermanentLockedAccountParams", { enumerable: true, get: function () { return tx_2.MsgCreatePermanentLockedAccount; } });
Object.defineProperty(exports, "MsgCreatePeriodicVestingAccountParams", { enumerable: true, get: function () { return tx_2.MsgCreatePeriodicVestingAccount; } });
/** MsgCreateVestingAccount defines a message that enables creating a vesting account. */
class MsgCreateVestingAccount {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/cosmos.vesting.v1beta1.MsgCreateVestingAccount",
                value: this.params,
                encode: () => tx_1.MsgCreateVestingAccount.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            throw new Error("MsgCreateVestingAccount not implemented.");
        });
    }
}
exports.MsgCreateVestingAccount = MsgCreateVestingAccount;
class MsgCreatePermanentLockedAccount {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/cosmos.vesting.v1beta1.MsgCreatePermanentLockedAccount",
                value: this.params,
                encode: () => tx_1.MsgCreatePermanentLockedAccount.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            throw new Error("MsgCreatePermanentLockedAccount not implemented.");
        });
    }
}
exports.MsgCreatePermanentLockedAccount = MsgCreatePermanentLockedAccount;
class MsgCreatePeriodicVestingAccount {
    constructor(params) {
        this.params = params;
    }
    toProto() {
        return __awaiter(this, void 0, void 0, function* () {
            return {
                type_url: "/cosmos.vesting.v1beta1.MsgCreatePeriodicVestingAccount",
                value: this.params,
                encode: () => tx_1.MsgCreatePeriodicVestingAccount.encode(this.params).finish(),
            };
        });
    }
    toAmino() {
        return __awaiter(this, void 0, void 0, function* () {
            throw new Error("MsgCreatePeriodicVestingAccount not implemented.");
        });
    }
}
exports.MsgCreatePeriodicVestingAccount = MsgCreatePeriodicVestingAccount;
